<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 8 | Pathway and Network Analysis 2024</title>
  <meta name="description" content="Workshop website for the 2024 Pathway and Network Analysis Canadian Bioinformatics Workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 8 | Pathway and Network Analysis 2024" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bioinformaticsdotca.github.io/PNA_2024/img/bioinformatics_logo.png" />
  <meta property="og:description" content="Workshop website for the 2024 Pathway and Network Analysis Canadian Bioinformatics Workshop" />
  <meta name="github-repo" content="bioinformaticsdotca/PNA_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 8 | Pathway and Network Analysis 2024" />
  
  <meta name="twitter:description" content="Workshop website for the 2024 Pathway and Network Analysis Canadian Bioinformatics Workshop" />
  <meta name="twitter:image" content="https://bioinformaticsdotca.github.io/PNA_2024/img/bioinformatics_logo.png" />

<meta name="author" content="Faculty: Veronique Voisin, Ruth Isserlin, Gary Bader, Lincoln Stein, Gregory Schwartz" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-7.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css">
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "name": "Pathway and Network Analysis",
  "description": "The complete set of tutorials, slides, and practical exercises for the Pathway and Network Analysis 2024 Canadian Bioinformatics Workshop, hosted on GitHub Pages.",
  "url": "https://bioinformaticsdotca.github.io/PNA_2024",
  "learningResourceType": ["tutorial", "slides", "exercise"],
  "inLanguage": "en",
  "license": "https://creativecommons.org/licenses/by/4.0/",
  "provider": {
    "@type": "Organization",
    "name": "Bioinformatics.ca",
    "url": "https://bioinformatics.ca"
  },
  "keywords": "",
  "isPartOf": {
    "@type": "Course",
    "@id": "https://bioinformatics.ca/workshops-all/2024-pathway-and-network-analysis-toronto/"
  }
}
</script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br>Pathway and Network Analysis<br><br></a></strong></center></li>
<a href="./">
 <img src="img/PNA_icon.png" width="70%" alt="Pathway and Network Analysis logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Workshop Info</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-work"><i class="fa fa-check"></i>Pre-work</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#class-photo"><i class="fa fa-check"></i>Class Photo</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="meet-your-faculty.html"><a href="meet-your-faculty.html"><i class="fa fa-check"></i>Meet Your Faculty</a></li>
<li class="part"><span><b>II Modules</b></span></li>
<li class="chapter" data-level="" data-path="module-1.html"><a href="module-1.html"><i class="fa fa-check"></i>Module 1</a>
<ul>
<li class="chapter" data-level="" data-path="module-1.html"><a href="module-1.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html"><i class="fa fa-check"></i>Module 2</a>
<ul>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#lecture-1"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#lab"><i class="fa fa-check"></i>Lab</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-1---launch-gprofiler."><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-3---adjust-parameters."><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-4---run-query"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-5---explore-the-results."><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-6-expand-the-stats-tab"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-7-save-the-results"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-8-optional-but-recommended"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-9-optional-by-recommended"><i class="fa fa-check"></i>Step 9 (Optional by recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query"><i class="fa fa-check"></i>Exercise 2: Load and use a custom .gmt file and run the query</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#optional-steps"><i class="fa fa-check"></i>Optional steps</a>
<ul>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#optional-1"><i class="fa fa-check"></i><strong>Optional 1</strong>:</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#option-2"><i class="fa fa-check"></i><strong>Option 2</strong>:</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#option-3"><i class="fa fa-check"></i><strong>Option 3</strong> :</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#bonus---automation."><i class="fa fa-check"></i>Bonus - Automation.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#background"><i class="fa fa-check"></i>Background</a>
<ul>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step3."><i class="fa fa-check"></i>Step3.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#bonus---automation.-1"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html"><i class="fa fa-check"></i>Module 3</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#lecture-2"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#lab-cytoscape-primer"><i class="fa fa-check"></i>Lab: Cytoscape primer</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-1a---create-network-from-table"><i class="fa fa-check"></i>Exercise 1a - Create Network from table</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-1b---load-node-attributes"><i class="fa fa-check"></i>Exercise 1b - Load node attributes</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-1c---map-node-attributes-to-visual-style"><i class="fa fa-check"></i>Exercise 1c - Map node attributes to Visual Style</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-2---work-with-larger-networks"><i class="fa fa-check"></i>Exercise 2 - Work with larger networks</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-3---perform-basic-enrichment-analysis-using-enrichmenttable"><i class="fa fa-check"></i>Exercise 3 - Perform basic enrichment analysis using EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#enrichmenttabl-features"><i class="fa fa-check"></i>Exercise 3B - create Enrichment Map and Enhanced graphics nodes from EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-4---load-network-from-ndex"><i class="fa fa-check"></i>Exercise 4 - Load network from NDex</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#lab-gprofiler-visualization"><i class="fa fa-check"></i>Lab: g:profiler Visualization</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#data-3"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-1a---compare-different-gprofiler-geneset-size-results"><i class="fa fa-check"></i>Exercise 1a - compare different gprofiler geneset size results</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-1b---is-specifying-the-gmt-file-important"><i class="fa fa-check"></i>Exercise 1b - Is specifying the gmt file important?</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-1c---create-em-from-results-using-baderlab-genesets"><i class="fa fa-check"></i>Exercise 1c - create EM from results using Baderlab genesets</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-1d-optional---investigate-individual-pathways-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 1d (optional) - investigate individual pathways in GeneMANIA or String</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#genemania"><i class="fa fa-check"></i>GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#string"><i class="fa fa-check"></i>String</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#bonus---automation.-2"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#lab-gsea-visualization"><i class="fa fa-check"></i>Lab: GSEA Visualization</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#data-4"><i class="fa fa-check"></i>Data</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-1---gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>Exercise 1 - GSEA output and EnrichmentMap</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network"><i class="fa fa-check"></i>Exercise 2 - Post analysis (add drug target gene-sets to the network)</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-3---autoannotate-the-network"><i class="fa fa-check"></i>Exercise 3 - Autoannotate the Network</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercise-4-optional---explore-results-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 4 (Optional) - Explore results in GeneMANIA or STRING</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#bonus---automation.-3"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#automation"><i class="fa fa-check"></i>Module 3 Lab: (Bonus) Automation</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#set-up---option-1---install-rrstudio"><i class="fa fa-check"></i>Set Up - Option 1 - Install R/Rstudio</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#set-up---option-2---docker-image-with-rrstudio"><i class="fa fa-check"></i>Set Up - Option 2 - Docker image with R/Rstudio</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#what-is-docker"><i class="fa fa-check"></i>What is docker?</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#docker---basic-term-definition"><i class="fa fa-check"></i>Docker - Basic term definition</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#container"><i class="fa fa-check"></i>Container</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#image"><i class="fa fa-check"></i>Image</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#docker-volumes"><i class="fa fa-check"></i>Docker Volumes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#r_docker"><i class="fa fa-check"></i>Install Docker</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#windows"><i class="fa fa-check"></i>Windows</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#macos-linux"><i class="fa fa-check"></i>MacOS / Linux</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#create-your-first-notebook-using-docker"><i class="fa fa-check"></i>Create your first notebook using Docker</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#start-coding"><i class="fa fa-check"></i>Start coding!</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#start-using-automation"><i class="fa fa-check"></i>Start using automation</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#running-example-notebooks-in-local-rstudio"><i class="fa fa-check"></i>Running example notebooks in local RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-3---open-example-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example RNotebook</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#step-4---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 4 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html"><i class="fa fa-check"></i>Module 4</a>
<ul>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#lecture-3"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#lab-1"><i class="fa fa-check"></i>Lab</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#goal-of-this-practical-lab"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#data-download-the-following-files-on-your-computer-before-starting-the-practical-lab."><i class="fa fa-check"></i>Data: download the following files on your computer before starting the practical lab.</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#exercise-1-use-the-reactome-functional-interaction-fi-network"><i class="fa fa-check"></i>Exercise 1: Use the Reactome Functional Interaction (FI) Network</a>
<ul>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#question-1-describe-the-size-and-composition-of-the-network"><i class="fa fa-check"></i>Question 1: Describe the size and composition of the network?</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#question-2-after-clustering-how-many-modules-are-there"><i class="fa fa-check"></i>Question 2: After clustering, how many modules are there?</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#query-information-about-the-interaction-between-2-genes"><i class="fa fa-check"></i>Query information about the interaction between 2 genes:</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#question-3-what-are-the-most-significant-pathways-in-each-module"><i class="fa fa-check"></i>Question 3: What are the most significant pathways in each module?</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#set-the-size-of-the-nodes-proportional-to-the-mutation-frequencies-in-each-cancer"><i class="fa fa-check"></i>Set the size of the nodes proportional to the mutation frequencies in each cancer</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#play-around-with-the-styles-change-transparency-and-colors"><i class="fa fa-check"></i>Play around with the styles: change transparency and colors</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#create-a-pie-chart"><i class="fa fa-check"></i>Create a pie chart</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#create-a-subnetwork"><i class="fa fa-check"></i>Create a subnetwork</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#fetch-cancer-drugs-on-the-created-subnetwork"><i class="fa fa-check"></i>Fetch Cancer drugs on the created subnetwork</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#save-the-network-as-an-image-for-publication"><i class="fa fa-check"></i>Save the network as an image for publication</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#exercise-2a-explore-reactome-pathways"><i class="fa fa-check"></i>Exercise 2a: Explore Reactome Pathways</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#exercise-2b-pathway-enrichment-analysis-using-a-simple-gene-list"><i class="fa fa-check"></i>Exercise 2b: Pathway enrichment analysis using a simple gene list</a>
<ul>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#question-1-what-are-the-most-significant-biological-pathways-based-on-the-fdr"><i class="fa fa-check"></i>Question 1: What are the most significant biological pathways based on the FDR?</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#answer-to-question-1"><i class="fa fa-check"></i>Answer to Question 1</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#exercise-2c-pathway-based-analysis-using-a-rank-gene-list-gsea"><i class="fa fa-check"></i>Exercise 2c: Pathway-based analysis using a rank gene list (GSEA)</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#automation-for-advanced-users"><i class="fa fa-check"></i>Automation ( for advanced users)</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#reference-guide-bonus-exercises"><i class="fa fa-check"></i>Reference guide /bonus exercises:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html"><i class="fa fa-check"></i>Module 5</a>
<ul>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#lecture-4"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#lab-genemania-cytoscape-version"><i class="fa fa-check"></i>Lab: GeneMANIA (Cytoscape version)</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#goal-of-this-practical-lab-1"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1-searching-genemania-with-single-gene"><i class="fa fa-check"></i>EXERCISE 1: Searching GeneMANIA with single gene</a>
<ul>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#answers"><i class="fa fa-check"></i>ANSWERS</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2-searching-genemania-with-gene-list"><i class="fa fa-check"></i>EXERCISE 2: Searching GeneMANIA with gene list</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3-searching-genemania-with-mixed-gene-list"><i class="fa fa-check"></i>EXERCISE 3: Searching GeneMANIA with mixed gene list</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#genemania-definitions"><i class="fa fa-check"></i>GeneMANIA DEFINITIONS:</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-4-optional-discover-the-stringapp"><i class="fa fa-check"></i>EXERCISE 4 (OPTIONAL): Discover the stringApp</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#more-string-information-and-tutorials"><i class="fa fa-check"></i>More STRING information and tutorials:</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#lab-genemania-web-version"><i class="fa fa-check"></i>Lab (GeneMANIA web version)</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#goal-of-this-practical-lab-2"><i class="fa fa-check"></i>Goal of this practical lab</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 1: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1-answers-detailed-explanation-and-screenshots"><i class="fa fa-check"></i>EXERCISE 1 ANSWERS: DETAILED EXPLANATION AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1---steps-1-4"><i class="fa fa-check"></i>EXERCISE 1 - STEPS 1-4</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1---step-5"><i class="fa fa-check"></i>EXERCISE 1 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1---step-6"><i class="fa fa-check"></i>EXERCISE 1 - STEP 6</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1---step-7"><i class="fa fa-check"></i>Exercise 1 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1---step-8"><i class="fa fa-check"></i>Exercise 1 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1---step-9"><i class="fa fa-check"></i>Exercise 1 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1---step-10-layouts"><i class="fa fa-check"></i>Exercise 1 - STEP 10 (layouts)</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-1---step-11-save-an-image"><i class="fa fa-check"></i>Exercise 1 - STEP 11 (save an image)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 2: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2-answers-detailed-steps-and-screenshots"><i class="fa fa-check"></i>EXERCISE 2 ANSWERS: DETAILED STEPS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---steps-1-to-4"><i class="fa fa-check"></i>Exercise 2 - STEPS 1 to 4</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-5"><i class="fa fa-check"></i>Exercise 2 - STEP 5</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-6."><i class="fa fa-check"></i>Exercise 2 - STEP 6.</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-7"><i class="fa fa-check"></i>Exercise 2 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-8"><i class="fa fa-check"></i>Exercise 2 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-9"><i class="fa fa-check"></i>Exercise 2 - STEP 9</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-10"><i class="fa fa-check"></i>Exercise 2 - STEP 10</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-11"><i class="fa fa-check"></i>Exercise 2 - STEP 11</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-12"><i class="fa fa-check"></i>Exercise 2 - STEP 12</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-2---step-13."><i class="fa fa-check"></i>Exercise 2 - STEP 13.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3-questions-and-steps-to-follow"><i class="fa fa-check"></i>EXERCISE 3: QUESTIONS AND STEPS TO FOLLOW</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3-more-details-and-screenshots"><i class="fa fa-check"></i>Exercise 3: MORE DETAILS AND SCREENSHOTS</a>
<ul>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3---steps-1---3"><i class="fa fa-check"></i>Exercise 3 - STEPS 1 - 3</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3---step-4-step5"><i class="fa fa-check"></i>Exercise 3 - STEP 4/ STEP5</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3---steps-6"><i class="fa fa-check"></i>Exercise 3 - STEPS 6</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3---step-7"><i class="fa fa-check"></i>Exercise 3 - STEP 7</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3---step-8"><i class="fa fa-check"></i>Exercise 3 - STEP 8</a></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#exercise-3---step-9"><i class="fa fa-check"></i>Exercise 3 - STEP 9</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5.html"><a href="module-5.html#some-definitions"><i class="fa fa-check"></i>SOME DEFINITIONS:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html"><i class="fa fa-check"></i>Module 6</a>
<ul>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#lecture-5"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#scrna-lab-praticals"><i class="fa fa-check"></i>scRNA lab praticals</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#lab-1-pmbc"><i class="fa fa-check"></i>Lab 1: PMBC</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#pmbc3k-seurat-pipeline"><i class="fa fa-check"></i>Pmbc3k Seurat Pipeline</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#load-libraries"><i class="fa fa-check"></i>load libraries</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#load-the-pbmc-dataset"><i class="fa fa-check"></i>Load the PBMC dataset</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#process-the-dataset"><i class="fa fa-check"></i>Process the dataset</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#assign-cell-type-identity-to-clusters"><i class="fa fa-check"></i>Assign cell type identity to clusters</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#find-differentially-expressed-features-cluster-biomarkers"><i class="fa fa-check"></i>Find differentially expressed features (cluster biomarkers)</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#create-gene-list-for-each-cluster-to-use-with-gprofiler"><i class="fa fa-check"></i>Create Gene list for each cluster to use with g:Profiler</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#tutorial_start"><i class="fa fa-check"></i>Data (gene lists for each cluster)</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#run-pathway-enrichment-analysis-using-gprofiler"><i class="fa fa-check"></i>Run pathway enrichment analysis using g:Profiler</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#create-an-enrichment-map-in-cytoscape"><i class="fa fa-check"></i>Create an enrichment map in Cytoscape</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#gsea-from-pseudobulk"><i class="fa fa-check"></i>GSEA from pseudobulk</a>
<ul>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#pseudobulk-creation-differential-expression-and-rank-file"><i class="fa fa-check"></i>pseudobulk creation, differential expression and rank file</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#run-gsea"><i class="fa fa-check"></i>run GSEA:</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#create-an-enrichmentmap"><i class="fa fa-check"></i>Create an EnrichmentMap:</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#lab-2-glioblastoma"><i class="fa fa-check"></i>Lab 2: Glioblastoma</a>
<ul>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#goal"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#data-5"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#overview"><i class="fa fa-check"></i>Overview</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#can-module8-exercise-1"><i class="fa fa-check"></i>Part 1 - run g:Profiler [OPTIONAL]</a>
<ul>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-1---launch-gprofiler.-1"><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-2---input-query-1"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-3---adjust-parameters.-1"><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-4---run-query-1"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-5---explore-the-results.-1"><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-6-expand-the-stats-tab-1"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-7-save-the-results-1"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-8-optional-but-recommended-1"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#exercise-2"><i class="fa fa-check"></i>Part 2 - Cytoscape/EnrichmentMap [OPTIONAL]</a>
<ul>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#goal-of-the-exercise-6"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#data-6"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#enrichmentmap-2"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#description-of-this-exercise-1"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#start-the-exercise-3"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-1-2"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-2-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-3-explore-detailed-results"><i class="fa fa-check"></i>Step 3: Explore Detailed results</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-4-optional-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Step 4 [OPTIONAL]: AutoAnnotate the enrichment map</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#exercise-3"><i class="fa fa-check"></i>Part 3 - Master map using multiple datasets</a>
<ul>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#goal-1"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#data-7"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#start-the-exercise-4"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-1-3"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#step-2-3"><i class="fa fa-check"></i>Step 2</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#lab-3-cellphonedb"><i class="fa fa-check"></i>Lab 3: cellPhoneDB</a>
<ul>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#presentation"><i class="fa fa-check"></i>Presentation</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#method"><i class="fa fa-check"></i>Method</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#examining-the-results"><i class="fa fa-check"></i>Examining the results</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#visualization-using-cytoscape"><i class="fa fa-check"></i>Visualization using Cytoscape</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#dataset-and-references"><i class="fa fa-check"></i>Dataset and references</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#dataset_prep"><i class="fa fa-check"></i>Dataset preprocessing and running CellPhoneDB</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#lab-4-nest"><i class="fa fa-check"></i>Lab 4: NEST</a>
<ul>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#nest-neural-network-on-spatial-transcriptomics"><i class="fa fa-check"></i>NEST (NEural network on Spatial Transcriptomics)</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#how-to-run-nest"><i class="fa fa-check"></i>How to run NEST</a></li>
<li class="chapter" data-level="" data-path="module-6.html"><a href="module-6.html#practical-lab-pancreatic-ductal-adenocarcinoma-pdac"><i class="fa fa-check"></i>Practical lab : Pancreatic Ductal Adenocarcinoma (PDAC)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html"><i class="fa fa-check"></i>Module 7</a>
<ul>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#lecture-slides"><i class="fa fa-check"></i>Lecture slides</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#lab-integrated-assignment"><i class="fa fa-check"></i>Lab: Integrated Assignment</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#goal-2"><i class="fa fa-check"></i>Goal</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#dataset-1"><i class="fa fa-check"></i>DATASET 1</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#background-1"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#data-processing"><i class="fa fa-check"></i>Data processing</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#part-1-run-gprofiler"><i class="fa fa-check"></i>PART 1: run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#part-2-save-as-generic-enrichment-map-output-be"><i class="fa fa-check"></i>PART 2: save as Generic Enrichment Map output (BE)</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#part-3-save-as-generic-enrichment-map-output-ne"><i class="fa fa-check"></i>PART 3: save as Generic Enrichment Map output (NE)</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#part-4-create-an-enrichment-map"><i class="fa fa-check"></i>PART 4: create an enrichment map</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#answers-gprofiler"><i class="fa fa-check"></i>Answers g:Profiler</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#part-5-gsea-run-and-create-an-enrichment-map"><i class="fa fa-check"></i>PART 5: GSEA (run and create an enrichment map)</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#part-6-iregulon"><i class="fa fa-check"></i>PART 6: iRegulon</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#dataset-2"><i class="fa fa-check"></i>DATASET 2</a></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#part-1-reactomefi"><i class="fa fa-check"></i>PART 1: ReactomeFI</a>
<ul>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#answers-reactome-fi"><i class="fa fa-check"></i>Answers REACTOME FI</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#part-2-genemania"><i class="fa fa-check"></i>PART 2: GeneMANIA</a>
<ul>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#answers-genemania"><i class="fa fa-check"></i>Answers GeneMANIA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-7.html"><a href="module-7.html#integrated-assignment-bonus---automation"><i class="fa fa-check"></i>Integrated Assignment Bonus - Automation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html"><i class="fa fa-check"></i>Module 8</a>
<ul>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#lecture-6"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#practical-lab-1-chip_seq-data---great-and-meme-chip"><i class="fa fa-check"></i>Practical lab 1: chIP_seq data - GREAT and MEME-chIP</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#practical-lab-2-gene-list---iregulon-and-enrichrenrichmentmap"><i class="fa fa-check"></i>Practical lab 2: gene list - iREgulon and enrichr/EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture"><i class="fa fa-check"></i>Additional slides about the tools Segway and BEHST presented during the lecture</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#optional-module-8-lab-1-gene-regulation-and-motif-analysis-practical-lab-chip-seq"><i class="fa fa-check"></i>Optional Module 8 Lab 1: Gene Regulation and Motif Analysis Practical Lab /chIP-seq</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#dataset-used-during-this-practical-lab"><i class="fa fa-check"></i>Dataset used during this practical lab</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#exercise-1---run-pathway-analysis-using-great"><i class="fa fa-check"></i>Exercise 1 - Run pathway analysis using GREAT</a>
<ul>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#perform-pathway-enrichment"><i class="fa fa-check"></i>Perform pathway enrichment</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#explore-the-results."><i class="fa fa-check"></i>Explore the results.</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#perform-pathway-enrichment---proximal-approach"><i class="fa fa-check"></i>Perform pathway enrichment - Proximal approach</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#explore-the-results.---proximal-analysis"><i class="fa fa-check"></i>Explore the results. - proximal analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#exercise-2---build-an-enrichment-map-to-visualize-great-results"><i class="fa fa-check"></i>Exercise 2 - Build an enrichment map to visualize GREAT results</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#exercise-3-optional-practice-building-enrichment-maps-and-auto-annotation"><i class="fa fa-check"></i>Exercise 3 (optional): Practice building enrichment maps and auto-annotation</a>
<ul>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#optional-exercise-3a-autoannotate-the-enrichment-map"><i class="fa fa-check"></i>Optional exercise 3a: AutoAnnotate the enrichment map:</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv."><i class="fa fa-check"></i>Optional exercise 3b: Repeat the process of building an enrichment map using the proximal data (Proximal_GOBP_greatExportAll.tsv).</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time."><i class="fa fa-check"></i>Optional exercise 3c: Repeat the process by building both the Proximal and Distal enrichment maps at the same time.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#exercise-4-add-runx1-targets-and-runx1-ko-genes-on-the-distal-enrichment-map."><i class="fa fa-check"></i>Exercise 4: Add RUNX1 targets and RUNX1 KO genes on the distal enrichment map.</a>
<ul>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#step-4a-post-analysis"><i class="fa fa-check"></i>step 4a: post analysis:</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#step-4b-optional-change-the-edge-style-of-the-signature-gene-sets"><i class="fa fa-check"></i>Step 4b Optional: Change the edge style of the signature gene-sets:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip"><i class="fa fa-check"></i>Exercise 5: Learning how to run MEME-chip from the MEME suite (https://meme-suite.org/meme/tools/meme-chip)</a>
<ul>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#format-the-data"><i class="fa fa-check"></i>Format the Data</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#exercise-5a-download-sequences-from-.bed-coordinates"><i class="fa fa-check"></i>Exercise 5a: Download sequences from .bed coordinates</a></li>
<li class="chapter" data-level="" data-path="module-8.html"><a href="module-8.html#exercise-5b-run-meme-chip"><i class="fa fa-check"></i>Exercise 5b: Run MEME-chIP</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>
<a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.png" height="55" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" height="55" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">
<br><br>
<a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=35% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo."><br>
<a href="https://computationalgenomics.ca/"><img src="img/sponsors/c3g-logo.png" width=90% style="padding-bottom: 20px" alt="Canadian Centre for Computational Genomics logo."></a><br>
<a href="https://www.hpcforhealth.ca/"><img src="img/sponsors/hpc4health.png" width=90% style="padding-bottom: 20px" alt="HPC4Health logo."></a>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway and Network Analysis 2024</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-8" class="section level1 hasAnchor">
<h1>Module 8<a href="module-8.html#module-8" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="lecture-6" class="section level2 hasAnchor">
<h2>Lecture<a href="module-8.html#lecture-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="./lectures/Pathways_2021_Module5_lecture_MH.pdf">Lecture slides</a></p>
<p><a href="https://www.youtube.com/watch?v=6rKCUOqGtXA&amp;list=PL3izGL6oi0S-xaoH8p9LnJD8RQm8eNWF2&amp;index=5">Recorded video</a></p>
</div>
<div id="practical-lab-1-chip_seq-data---great-and-meme-chip" class="section level2 hasAnchor">
<h2>Practical lab 1: chIP_seq data - GREAT and MEME-chIP<a href="module-8.html#practical-lab-1-chip_seq-data---great-and-meme-chip" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="./lectures/Pathways_2021_Module5_practical_lab_CHIPseq_lab_vv.pdf">chIP_seq Lab slides</a></p>
<p><a href="#regulatory_network_chipseq_lab">chIP_seq Lab practical</a></p>
</div>
<div id="practical-lab-2-gene-list---iregulon-and-enrichrenrichmentmap" class="section level2 hasAnchor">
<h2>Practical lab 2: gene list - iREgulon and enrichr/EnrichmentMap<a href="module-8.html#practical-lab-2-gene-list---iregulon-and-enrichrenrichmentmap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="./lectures/Pathways_2021_Module5lab_iregulon.pdf">iREgulon Lab slides</a></p>
<p><a href="#regulatory_network_lab">iREgulon Lab practical</a></p>
</div>
<div id="additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture" class="section level2 hasAnchor">
<h2>Additional slides about the tools Segway and BEHST presented during the lecture<a href="module-8.html#additional-slides-about-the-tools-segway-and-behst-presented-during-the-lecture" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="./lectures/Pathways_2021_Segway_GMTK02_UTMIST_2021.pdf">Segway slides</a><br />
<a href="./lectures/Pathways_2021_segway_semi_automated_genome_annotation_post_submission_draft.pdf">Segway protocol_draft</a></p>
<p><a href="./lectures/Pathways_2021_BEHST07_Asilomar_Chromatin_2020.pdf">BEHST slides</a></p>
</div>
<div id="optional-module-8-lab-1-gene-regulation-and-motif-analysis-practical-lab-chip-seq" class="section level2 hasAnchor">
<h2>Optional Module 8 Lab 1: Gene Regulation and Motif Analysis Practical Lab /chIP-seq<a href="module-8.html#optional-module-8-lab-1-gene-regulation-and-motif-analysis-practical-lab-chip-seq" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Perform pathway analysis starting with a chIP_seq bed file and visualize the results using Cytoscape/EnrichmentMap.</li>
<li>Be able to use the tool GREAT with distal and proximal parameters.</li>
<li>Run MEME-chip to find over-enrichment of transcription factors.</li>
<li>Optional: learn how to use iRegulon to find targets of a transcription factor of interest and find orthologs using the tool g:Profiler/g:orth.</li>
</ul>
<p>This practical lab consists of 6 exercises and 2 of them are optional. Follow the step-by-step checklist through the exercises.</p>
<p>Before starting the lab, download the files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select Save Link As.
</p>
<p>
Place the file in your CBW work directory in the corresponding module
directory.
</p>
</div>
<ul>
<li><a href="./Module5/chipseqlab/chipseqlab_data/GSE128767_RUNX1_ChIP.peaks.bed">GSE128767_RUNX1_ChIP.peaks.bed</a></li>
<li><a href="./Module5/chipseqlab/chipseqlab_data/Distal_GOBP_greatExportAll.tsv">Distal_GOBP_greatExportAll.tsv</a></li>
<li><a href="./Module5/chipseqlab/chipseqlab_data/Proximal_GOBP_greatExportAll.tsv">Proximal_GOBP_greatExportAll.tsv</a></li>
<li><a href="./Module5/chipseqlab/chipseqlab_data/RUNX1_Affy.gmt">RUNX1_Affy.gmt</a></li>
<li><a href="./Module5/chipseqlab/chipseqlab_data/GSE128767_RUNX1_ChIP.peaks.fasta">GSE128767_RUNX1_ChIP.peaks.fasta</a></li>
</ul>
<div class="rmd-note">
<p>
EnrichmentMap and Cytoscape layouts: Network layouts are flexible and
can be rearranged. What you see when you perform these exercises may not
be identical in appearance to what you see in the screenshots in the
practical lab, or what you have seen other times that you have performed
the exercises.
</p>
</div>
</div>
<div id="dataset-used-during-this-practical-lab" class="section level2 hasAnchor">
<h2>Dataset used during this practical lab<a href="module-8.html#dataset-used-during-this-practical-lab" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>ChIP-seq for RUNX1 from pools of mouse CD1 fetal ovaries (E14.5)<br>
NCBI GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE128767">GSE128767</a></p>
<p>From the paper:<br> <strong>RUNX1 maintains the identity of the fetal ovary through an interplay with FOXL2</strong><br> Nicol B, Grimm SA, Chalmel F, Lecluze E et al.<br><a href="https://www.nature.com/articles/s41467-019-13060-1">Nat Commun 2019 Nov 11;10(1):5116</a>.<br> <a href="https://pubmed.ncbi.nlm.nih.gov/31712577/">PMID: 31712577</a></p>
<p><strong>Abstract</strong>:<br> Sex determination of the gonads begins with <strong>fate specification</strong> of gonadal supporting cells into either ovarian granulosa cells or testicular Sertoli cells. This process of fate specification hinges on a balance of transcriptional control. We discovered that the <strong>transcription factor RUNX1</strong> is enriched in the <strong>fetal ovary</strong> in rainbow trout, turtle, mouse, and human. In the mouse, RUNX1 marks the supporting cell lineage and becomes granulosa cell-specific as the gonads differentiate. RUNX1 plays complementary/redundant roles with FOXL2 to maintain fetal granulosa cell identity, and combined loss of RUNX1 and FOXL2 results in masculinization of the fetal ovaries. To determine whether interplay between RUNX1 and FOXL2 occurs at the chromatin level, <strong>we performed genome-wide analysis of RUNX1 chromatin occupancy in E14.5 ovaries. The top de novo motif identified in RUNX1 ChIP-seq matched the RUNX motif</strong>. We found that RUNX1 chromatin occupancy was partially overlapping with FOXL2 chromatin occupancy in fetal ovaries.</p>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img2.png" alt="Figure 1" />
<div class="figcaption">Figure 1</div>
</div>
<p>They found that RUNX1 is expressed in the fetal ovary at day 14 in mice and that it is necessary for a good development of the ovary.</p>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img3.png" alt="Figure 2" />
<div class="figcaption">Figure 2</div>
</div>
<p>A KO of Runx1 and another TF Foxl2 abolished the normal development of the ovary.</p>
<p>#Why did we choose this dataset?</p>
<ul>
<li>RUNX1 is a transcription factor that is interesting to study as it has major biological functions.</li>
<li>chIP-seq peaks are stored in a bed file that can be download from GEO entry.</li>
<li>Linked to transcriptomic data <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE129038">GSE129038</a></li>
<li>Quality of the data</li>
</ul>
<p>#The 3 pieces of information that we need to get before starting the analysis are:</p>
<ul>
<li>the model organism: Mus musculus</li>
<li>genome version: mm10</li>
<li>bed file : <a href="./Module5/chipseqlab/chipseqlab_data/GSE128767_RUNX1_ChIP.peaks.bed">GSE128767_RUNX1_ChIP.peaks.bed</a></li>
</ul>
<p>We have indicated below how we retrieved these information <strong>but you dont need to do it for the lab</strong>:</p>
<ul>
<li>In the main GEO entry <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE128767">GSE128767</a></li>
<li>click on one of the samples (for example - <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3684638">GSM3684638</a>).<br />
</li>
<li>On the sample page scroll down to the Data processing section</li>
<li>The organism is <strong>mus musculus</strong> and the reference genome is <strong>mm10</strong></li>
<li>3 files are available from the GEO entry (see below).</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img4.png" alt="Figure 3 - Dataset BED file" />
<div class="figcaption">Figure 3 - Dataset BED file</div>
</div>
<ul>
<li>The bed file provided by the authors (GSE128767_RUNX1_ChIP.peaks.bed) (linked on the main dataset page under supplementary file - <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE128767">GSE128767</a> )
has the right format to be used by <a href="http://great.stanford.edu/public/html/">GREAT</a> for the pathway analysis;
The first 3 fields contain the chromosome name, start and end. They are the 3 required fields. The fourth column is optional and consists of the chromosomal position, followed by MACS2 score value and FDR. Below is an example view of the BED file:</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img5.png" alt="Figure 4 - Example view of BED file" />
<div class="figcaption">Figure 4 - Example view of BED file</div>
</div>
</div>
<div id="exercise-1---run-pathway-analysis-using-great" class="section level2 hasAnchor">
<h2>Exercise 1 - Run pathway analysis using GREAT<a href="module-8.html#exercise-1---run-pathway-analysis-using-great" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="perform-pathway-enrichment" class="section level3 hasAnchor">
<h3>Perform pathway enrichment<a href="module-8.html#perform-pathway-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Open a web browser and open <a href="http://great.stanford.edu/public/html/">GREAT</a></li>
<li>In <strong>Species Assembly</strong> choose Mouse: GRCm38(UCSC mm10, Dec.2011)</li>
<li>In<strong>Test regions</strong> Click on <strong>Choose file</strong> and locate the file GSE128767_RUNX1_ChIP.peaks.bed that you save on your computer.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/%20img7.png" alt="Figure 5 - GREAT interface" />
<div class="figcaption">Figure 5 - GREAT interface</div>
</div>
<ul>
<li>In <strong>Association rule settings</strong>, click on <strong>Show settings</strong> to see the current rule set to associate genes to peaks</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img8.png" alt="Figure 6 - GREAT Association rules used in analysis" />
<div class="figcaption">Figure 6 - GREAT Association rules used in analysis</div>
</div>
<ul>
<li><strong>Do not change the settings.</strong> We are doing a distal analysis where genes (genomic regions) are associated with peaks within 5k upstream of the transcription start site of the genes (TSS), 1kb downstream and up to 1000 kb to nearest gene.</li>
<li>Click on the <strong>Submit</strong> button at the end of the page</li>
</ul>
</div>
<div id="explore-the-results." class="section level3 hasAnchor">
<h3>Explore the results.<a href="module-8.html#explore-the-results." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Expand the <strong>Job Description</strong> tab to check the parameters,</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img10.png" alt="Figure 7 - Job Description" />
<div class="figcaption">Figure 7 - Job Description</div>
</div>
<ul>
<li>click on <strong>View all genomic region-gene associations</strong> (second line from bottom in blue font)</li>
<li>In a new tab there will be 2 tables containing the list of the chIP-seq peaks and corresponding associated genes.</li>
<li>Download both of the tables (region -&gt; gene and gene -&gt; region)</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img10b.png" alt="Figure 8 - genomic region-gene association tables." />
<div class="figcaption">Figure 8 - genomic region-gene association tables.</div>
</div>
<ul>
<li>Return to the main GREAT results page.</li>
<li>In the main results tab, under the <strong>Region-Gene Association Graphs</strong>, we can see that the peaks were mainly associated with genes located +-5kb of the TSS in addition to the presence of some distal peaks as expected based on the association rule that we have used.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/distal.png" alt="Figure 9 - Region-gene association graphs" />
<div class="figcaption">Figure 9 - Region-gene association graphs</div>
</div>
<ul>
<li><p>Lets explore the pathway analysis results and look at the GO Biological Process table.</p></li>
<li><p>Scroll down to the <strong>GO Biological Process</strong> section.</p></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/distal_true2.png" alt="Figure 10 - GO Biological Process results" />
<div class="figcaption">Figure 10 - GO Biological Process results</div>
</div>
<p>As we defined a distal rule to associate peaks with genes, we are going to look at the <strong>binomial FDR</strong>. The binomial test assesses whether the number of genomic regions annotated with the tested pathway is significantly larger than the number of genomic regions not annotated with the tested pathway.
The fold enrichment is the proportion of genomic regions annotated with the tested pathway and genomic regions not annotated with the tested pathway.</p>
<ul>
<li>Export the GO BP result on your local computer:
<ul>
<li>Under the <strong>GO Biological Process</strong> title, locate the <strong>Table controls:</strong></li>
<li>select the option <strong>All ontology data as .tsv</strong>.</li>
<li>A file called greatExportAll.tsv will be saved on your computer.</li>
<li>Rename the file <strong>Distal_GOBP_greatExportAll.tsv</strong>. We will import this file later in Cytoscape/EnrichmentMap.</li>
</ul></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img13.png" alt="Figure 11 - Download Go Biological Process results" />
<div class="figcaption">Figure 11 - Download Go Biological Process results</div>
</div>
</div>
<div id="perform-pathway-enrichment---proximal-approach" class="section level3 hasAnchor">
<h3>Perform pathway enrichment - Proximal approach<a href="module-8.html#perform-pathway-enrichment---proximal-approach" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We are now trying a proximal approach to define genes associated with peaks.</p>
<ul>
<li>Go back to the main GREAT page. Make sure the bed file is still uploaded and the genome is set to mm10.</li>
<li>Locate the <strong>Association rule settings</strong> and click on <strong>Show settings</strong>.</li>
<li>Set Proximal 1kb upstream, 1kb downstream plus Distal up to 1kb .</li>
<li>Uncheck the <strong>Include curated regulatory domains</strong> box.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img9.png" alt="Figure 12 - GREAT Association rules used in proximal analysis" />
<div class="figcaption">Figure 12 - GREAT Association rules used in proximal analysis</div>
</div>
<ul>
<li>Click on Submit.</li>
</ul>
</div>
<div id="explore-the-results.---proximal-analysis" class="section level3 hasAnchor">
<h3>Explore the results. - proximal analysis<a href="module-8.html#explore-the-results.---proximal-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>In the <strong>Region-Gene Association Graphs</strong> , we can see that using the proximal rule in our settings, genes are associated with peaks that are all within the +-5kb rule (in fact the +-1kb rule) and there are no more distal peaks.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/proximal.png" alt="Figure 13 - Proximal Region-gene association graphs" />
<div class="figcaption">Figure 13 - Proximal Region-gene association graphs</div>
</div>
<ul>
<li>Explore the GOBP results and export the results on your computer.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/proximal2.png" alt="Figure 14 - Proximal GO BP results" />
<div class="figcaption">Figure 14 - Proximal GO BP results</div>
</div>
<p>Using this rule, genes will be associated with peaks only if they are within 1kb of the transcription start site of the genes. It reduces the problem to a gene list and in this case, a Fishers exact (Hyper FDR Q-Val) test can be applied to test for pathway enrichment.</p>
<ul>
<li>Export the GO BP result on your local computer:
<ul>
<li>Under the <strong>GO Biological Process</strong> title, locate the <strong>Table controls:</strong></li>
<li>select the option <strong>All ontology data as .tsv</strong>.</li>
<li>A file called greatExportAll.tsv will be saved on your computer.</li>
<li>Rename the file <strong>Proximal_GOBP_greatExportAll.tsv</strong>. We will import this file later in Cytoscape/EnrichmentMap.</li>
</ul></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img13.png" alt="Figure 15 - Export Proximal GOBP enrichment results" />
<div class="figcaption">Figure 15 - Export Proximal GOBP enrichment results</div>
</div>
</div>
</div>
<div id="exercise-2---build-an-enrichment-map-to-visualize-great-results" class="section level2 hasAnchor">
<h2>Exercise 2 - Build an enrichment map to visualize GREAT results<a href="module-8.html#exercise-2---build-an-enrichment-map-to-visualize-great-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Open Cytoscape</li>
<li>In the menu bar, select <strong>Apps</strong> &gt; <strong>EnrichmentMap</strong></li>
<li>Drag and drop the GREAT result file <strong>Distal_GOBP_greatExportAll.tsv</strong> into the DataSet box.</li>
<li>Set the FDR q value cut-off to <strong>0.001</strong></li>
<li>Click on <strong>Build</strong></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img14.png" alt="Figure 16 - Enrichment map input panel" />
<div class="figcaption">Figure 16 - Enrichment map input panel</div>
</div>
<ul>
<li>A <strong>Set Parameters</strong> dialog box opens: Choose <strong>Binomial p-value</strong>.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/img15.png" alt="Figure 17 - Statistical Test choice panel" />
<div class="figcaption">Figure 17 - Statistical Test choice panel</div>
</div>
<ul>
<li>Explore the map.</li>
</ul>
<p><img src="Module5/chipseqlab/chipseqlab_image/distal_map.png" alt="Figure 18 - Enrichment map with distal enriched pathways" />
* Note that EnrichmentMap creates annotations by default. You can delete the auto-generated annotation by going to the AutoAnnotate tab. Click on the settings menu (three lines) -&gt; Delete. Now you can create new annotation as explained below.</p>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/distalmap_AAsettings1.png" alt="Figure 18a - Enrichment map with auto-annotations" />
<div class="figcaption">Figure 18a - Enrichment map with auto-annotations</div>
</div>
</div>
<div id="exercise-3-optional-practice-building-enrichment-maps-and-auto-annotation" class="section level2 hasAnchor">
<h2>Exercise 3 (optional): Practice building enrichment maps and auto-annotation<a href="module-8.html#exercise-3-optional-practice-building-enrichment-maps-and-auto-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="optional-exercise-3a-autoannotate-the-enrichment-map" class="section level3 hasAnchor">
<h3>Optional exercise 3a: AutoAnnotate the enrichment map:<a href="module-8.html#optional-exercise-3a-autoannotate-the-enrichment-map" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>In the menu bar, select Apps and then AutoAnnotate.</li>
<li>A dialog box opens.</li>
<li>Go to Advanced. Under <strong>Cluster Creation</strong>, select <strong>Use clusterMaker2 App</strong></li>
<li>Under <strong>Label Creation</strong>, select <strong>Name of most significant node in cluster</strong></li>
<li>Click on <strong>Create Annotations</strong>.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/distalmap_AAsettings2.png" alt="Figure 19 - Autoannotate panel" />
<div class="figcaption">Figure 19 - Autoannotate panel</div>
</div>
<p>Arrange the display by clicking on each module name listed in the right panel and then move them apart from the other modules using a mouse or a trackpad.</p>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/distal_map_AA.png" alt="Figure 20 - Manually layed out Enrichment map of enriched pathways for distal set" />
<div class="figcaption">Figure 20 - Manually layed out Enrichment map of enriched pathways for distal set</div>
</div>
<div class="rmd-question">
<p>
What are the main biological functions enriched in genes associated
with RUNX1 peaks? <br> Is it relevant in relation to what we know about
the role of RUNX1 in development?
</p>
</div>
</div>
<div id="optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv." class="section level3 hasAnchor">
<h3>Optional exercise 3b: Repeat the process of building an enrichment map using the proximal data (Proximal_GOBP_greatExportAll.tsv).<a href="module-8.html#optional-exercise-3b-repeat-the-process-of-building-an-enrichment-map-using-the-proximal-data-proximal_gobp_greatexportall.tsv." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Because this is proximal data, the problem is reduced to a gene list and you can use the Fishers exact test (FDR 0.001) to looked at the enrichment results</p>
</div>
<div id="optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time." class="section level3 hasAnchor">
<h3>Optional exercise 3c: Repeat the process by building both the Proximal and Distal enrichment maps at the same time.<a href="module-8.html#optional-exercise-3c-repeat-the-process-by-building-both-the-proximal-and-distal-enrichment-maps-at-the-same-time." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Drag both files in the EnrichmentMap input box.</li>
<li>Use FDR 0.0001 for both and binomial test.</li>
<li>Check which nodes are in common between the 2 datasets.<br />
</li>
<li>Color the data by datasets.</li>
</ul>
</div>
</div>
<div id="exercise-4-add-runx1-targets-and-runx1-ko-genes-on-the-distal-enrichment-map." class="section level2 hasAnchor">
<h2>Exercise 4: Add RUNX1 targets and RUNX1 KO genes on the distal enrichment map.<a href="module-8.html#exercise-4-add-runx1-targets-and-runx1-ko-genes-on-the-distal-enrichment-map." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>During this exercise, we will connect the proximal chIP-seq enrichment map with the RUNX1 targets as well as the genes that are dysregulated after RUNX1 KO. We have already created a .gmt file that contains these gene lists (RUNX1_Affy.gmt). The format of a .gmt file is a tab delimited text file with one row per gene-set. Each gene-set contains the name of the gene-set, a description of the gene-set followed by the names of the genes. The file extension is changed from .txt to .gmt.</p>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/gmt.png" alt="Figure 20 - example of gmt file" />
<div class="figcaption">Figure 20 - example of gmt file</div>
</div>
<ul>
<li>Note: We extracted the RUNX1 targets using the iRegulon Cytoscape app and the optional exercise 6 describes the steps. We extracted 200 genes to build the RUNX1 target gene list.</li>
</ul>
<p>This RUNX1 study had transcriptomics data (microarray) in addition to the chIP-seq data. The microarray data gives an overview of all genes that are changing between a fetal ovary with normal development and a fetal ovary after RUNX1 knock-out (KO) (GSE129038). We have used the tool GEO2R to get the top 500 up and down regulated genes (see description of the steps at the end the document).</p>
<div id="step-4a-post-analysis" class="section level3 hasAnchor">
<h3>step 4a: post analysis:<a href="module-8.html#step-4a-post-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Go to the EnrichmentMap tab</li>
<li>Make sure that the Distal_GOBP_greatExportAll network is selected.</li>
<li>click on <strong>Options</strong> &gt; <strong>Add Signature Gene Sets</strong>.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/PA01.png" alt="Figure 21 - Add Signature sets" />
<div class="figcaption">Figure 21 - Add Signature sets</div>
</div>
<ul>
<li>Click on Load from File. located on the right hand size and select the file RUNX1_Affy.gmt that you have saved on your computer.</li>
<li>Set Test to Hypergeometric Test with the Cutoff set to 0.05.</li>
<li>Click on <strong>finish</strong></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/PA02.png" alt="Figure 22 - Signature sets input panel" />
<div class="figcaption">Figure 22 - Signature sets input panel</div>
</div>
<p>The 3 gene-sets are now added to the map. Each line (edge) shows pathways that have genes in common with the signature gene-sets.</p>
</div>
<div id="step-4b-optional-change-the-edge-style-of-the-signature-gene-sets" class="section level3 hasAnchor">
<h3>Step 4b Optional: Change the edge style of the signature gene-sets:<a href="module-8.html#step-4b-optional-change-the-edge-style-of-the-signature-gene-sets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Click on one signature gene-set node on the map to select it (it should appear in yellow).</li>
<li>In the Cytoscape menu bar, click Select &gt; Edges &gt;Select Adjacent Edges</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/pa1.png" alt="Figure 23 - Select adjacent edges" />
<div class="figcaption">Figure 23 - Select adjacent edges</div>
</div>
<ul>
<li>Go to Style and in the Edge table, next to <strong>Stroke Color (Unselected)</strong> click in the bypass column Byp. , click on the box and select a color.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/pa2.png" alt="Figure 24 - Bypass selected edge color" />
<div class="figcaption">Figure 24 - Bypass selected edge color</div>
</div>
<ul>
<li>Repeat for all genes:
<ul>
<li>In  Style and in the Edge table, go to Width and set Column to EM k_Intersection</li>
</ul></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/em3.png" alt="Figure 25 - Final figure" />
<div class="figcaption">Figure 25 - Final figure</div>
</div>
</div>
</div>
<div id="exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip" class="section level2 hasAnchor">
<h2>Exercise 5: Learning how to run MEME-chip from the MEME suite (<a href="https://meme-suite.org/meme/tools/meme-chip" class="uri">https://meme-suite.org/meme/tools/meme-chip</a>)<a href="module-8.html#exercise-5-learning-how-to-run-meme-chip-from-the-meme-suite-httpsmeme-suite.orgmemetoolsmeme-chip" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="format-the-data" class="section level3 hasAnchor">
<h3>Format the Data<a href="module-8.html#format-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>MEME suite accepts sequences as input and not chromosome coordinates. The bed file contains the chromosome coordinates of the peaks. Therefore, we first need to fetch all the peak sequences. UCSC genome browser (<a href="https://genome.ucsc.edu/" class="uri">https://genome.ucsc.edu/</a>) has some tools to help us.</p></li>
<li><p>If needed, you can use the finalized formatted file <a href="./Module5/chipseqlab/chipseqlab_data/GSE128767_RUNX1_ChIP.peaks.fasta">GSE128767_RUNX1_ChIP.peaks.fasta</a> to run MEME-chIP <strong>but we encourage you to follow the below steps to learn how to do it yourself</strong>.</p></li>
<li><p>The step that we took to create it are described below and were adapted from <a href="https://fasta.bioch.virginia.edu/cshl/stubbs/meme-ex/meme.html" class="uri">https://fasta.bioch.virginia.edu/cshl/stubbs/meme-ex/meme.html</a>.</p></li>
</ul>
</div>
<div id="exercise-5a-download-sequences-from-.bed-coordinates" class="section level3 hasAnchor">
<h3>Exercise 5a: Download sequences from .bed coordinates<a href="module-8.html#exercise-5a-download-sequences-from-.bed-coordinates" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Open the UCSC browser main page (<a href="http://genome.ucsc.edu/" class="uri">http://genome.ucsc.edu/</a>).</li>
<li>Click on <em>Genomes</em> in the menu bar and select <em>Mouse GRCm38/mm10</em>.</li>
</ul>
<p align="center">
<img src="Module5/chipseqlab/chipseqlab_image/meme1.png" alt="USCS main page" width="50%" />
</p>
<ul>
<li>The UCSC Genome Browser window opens in a new tab.</li>
<li>Below the tracks, click on the button <em>add custom tracks</em>. A new window will open.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/meme2.png" alt="UCSC genome browser" />
<div class="figcaption">UCSC genome browser</div>
</div>
<ul>
<li>Upload the bed file <a href="./Module5/chipseqlab/chipseqlab_data/GSE128767_RUNX1_ChIP.peaks.bed">GSE128767_RUNX1_ChIP.peaks.bed</a>; press the <strong>Submit</strong> button.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/meme3.png" alt="meme3" />
<div class="figcaption">meme3</div>
</div>
<ul>
<li>A new window will appear with your updated track. Make sure that <strong>Table Browser</strong> is selected and click on <em>go</em>.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/meme4.png" alt="meme4" />
<div class="figcaption">meme4</div>
</div>
<ul>
<li><p>A new window will appear. Select <em>sequence</em> as <em>output format</em> and <em>plain text</em> as <em>file type returned</em>. Click on <em>get output</em>.
<img src="Module5/chipseqlab/chipseqlab_image/meme5.png" alt="meme5" /></p></li>
<li><p>A new window will open where you can choose various options for your sequence (e.g.repeat masking). Note that for meme and similar programs it is important to <strong>mask repeats</strong> to <strong>N</strong>; otherwise, sequences in repetitive elements will dominate your motif list.</p>
<ul>
<li>Select <em>Mask repeats</em></li>
<li>next to <em>Mask repeats</em> change option to <em>to N</em></li>
<li>click on <em>Get sequences</em></li>
</ul></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/meme6.png" alt="meme6" />
<div class="figcaption">meme6</div>
</div>
<ul>
<li><p>A fasta file will appear; save this as plain text (copy and paste in a text editor or right click on the page and select <em>Save As</em> and save the file to your computer).</p>
<ul>
<li>here is the file in case you need it: <a href="./Module5/chipseqlab/chipseqlab_data/GSE128767_RUNX1_ChIP.peaks_INTERMEDIATE.fasta">GSE128767_RUNX1_ChIP.peaks_INTERMEDIATE.fasta</a></li>
</ul></li>
<li><p>You will need to modify the UCSC header that comes with the sequences to use them for meme:</p>
<ul>
<li>Go to <a href="https://fasta.bioch.virginia.edu/fasta_www2/clean_fasta.html" class="uri">https://fasta.bioch.virginia.edu/fasta_www2/clean_fasta.html</a></li>
<li>upload or copy and paste the plain text file from the above step</li>
<li>check Extract CHR:coordinates from UCSC</li>
<li>Click on Clean Sequence
<img src="Module5/chipseqlab/chipseqlab_image/meme7.png" alt="meme7" /></li>
<li>Save this as plain text under the name GSE128767_RUNX1_ChIP.peaks.fasta (copy and paste in a text editor or <strong>right click and Save as will not work for this file</strong>) - it will look like the below file.</li>
</ul></li>
</ul>
<p align="center">
<img src="Module5/chipseqlab/chipseqlab_image/meme8.png" alt="resulting file" width="50%" />
</p>
</div>
<div id="exercise-5b-run-meme-chip" class="section level3 hasAnchor">
<h3>Exercise 5b: Run MEME-chIP<a href="module-8.html#exercise-5b-run-meme-chip" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Open <a href="https://meme-suite.org/meme/tools/meme-chip" class="uri">https://meme-suite.org/meme/tools/meme-chip</a>.</li>
<li>Expand <em>Motif Discovery</em></li>
<li>Click on <em>MEME-Chip</em></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/meme9.png" alt="meme9" />
<div class="figcaption">meme9</div>
</div>
<ul>
<li>Under <em>Input the primary sequences</em> box, upload the file <a href="./Module5/chipseqlab/chipseqlab_data/GSE128767_RUNX1_ChIP.peaks.fasta">GSE128767_RUNX1_ChIP.peaks.fasta</a> .</li>
<li>Click on <em>Start Search</em>.</li>
</ul>
<p align="center">
<p><img src="Module5/chipseqlab/chipseqlab_image/meme10.png" alt="resulting file" width="75%" /></p>
<div class="rmd-note">
<p>
<strong>Important</strong>: Save the url so you can access your
result later even if you close the MEME window. <br>
</p>
<p>
For example my url is -
<a href="https://meme-suite.org/meme/info/status?service=MEMECHIP&amp;id=appMEMECHIP_5.3.31620409506563-973419203" class="uri">https://meme-suite.org/meme/info/status?service=MEMECHIP&amp;id=appMEMECHIP_5.3.31620409506563-973419203</a>
<br>
</p>
<p align="center">
<p>
<img src="Module5/chipseqlab/chipseqlab_image/meme11.png" alt="resulting file" width="50%" />
</p>
</div>
<ul>
<li>MEME-ChIP will run for about 1 hour:
<ul>
<li>look at the results below from the MEME-chip result,</li>
<li>try to answer the questions and follow next steps.</li>
<li>Check your MEME-ChIP results at the end of the practical lab.</li>
</ul></li>
<li>When your job is complete you should see the following page on your saved link:</li>
</ul>
<p align="center">
<p><img src="Module5/chipseqlab/chipseqlab_image/meme12.png" alt="jobs results page" width="50%" /></p>
<ul>
<li>results of the top motifs that were found significantly enriched in the peak sequences.<br></li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/meme13.png" alt="meme13" />
<div class="figcaption">meme13</div>
</div>
<div class="rmd-question">
<p>
To which transcription factor does it correspond? <br> Why is the
centered distribution of the motif important (what does it mean)?
</p>
</div>
<div id="exercise-6-optional-get-the-iregulon-runx1-targets-and-find-the-mouse-orthologs-using-gorth-from-gprofiler-to-create-the-gmt-file-used-in-exercise-4." class="section level2 hasAnchor">
<h2>Exercise 6 (optional): Get the iRegulon RUNX1 targets and find the mouse orthologs using g:Orth (from g:Profiler) to create the gmt file used in Exercise 4.<a href="module-8.html#exercise-6-optional-get-the-iregulon-runx1-targets-and-find-the-mouse-orthologs-using-gorth-from-gprofiler-to-create-the-gmt-file-used-in-exercise-4." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>In Cytoscape, locate App in the menu bar and select iRegulon and then Query TF-target database</li>
</ul>
<p align="center">
<p><img src="Module5/chipseqlab/chipseqlab_image/iregulon1.png" alt="iregulon" width="50%" /></p>
<ul>
<li>A Query TF-target database for a factor dialog box opens.
<ul>
<li>Enter RUNX1 in the <em>Transcription Factor</em> field and</li>
<li>in <em>Network</em>, set Number nodes (approx.) to 200.</li>
<li>Click on <em>Submit</em></li>
</ul></li>
</ul>
<p align="center">
<p><img src="Module5/chipseqlab/chipseqlab_image/iregulon2.png" alt="iregulon" width="75%" /></p>
<ul>
<li>To arrange the style,
<ul>
<li>go to the Cytoscape menu bar and select <em>Layout</em> &gt; <em>yFiles Organic Layout</em>.</li>
<li>Go the Cytoscape menu and select <em>View</em> &gt; <em>Always Show Graphic Details</em> to see the gene names.</li>
</ul></li>
<li>Below the network in the Table Panel:
<ul>
<li>click on <em>Node Table</em> and</li>
<li>click on the <em>Export Table to File</em> icon.</li>
<li>Click on <em>OK</em>.</li>
</ul></li>
</ul>
<p align="center">
<p><img src="Module5/chipseqlab/chipseqlab_image/iregulon3.png" alt="iregulon" width="75%" /></p>
<ul>
<li><p>A File <em>Metatargetome for RUNX1_1 default node.csv</em> is now saved to your computer.</p></li>
<li><p>Open the file <em>Metatargetome for RUNX1_1 default node.csv</em> and</p>
<ul>
<li>copy the gene list.</li>
<li>Open g:Profiler/g:orth at <a href="https://biit.cs.ut.ee/gprofiler/orth" class="uri">https://biit.cs.ut.ee/gprofiler/orth</a>.<br />
</li>
<li>Paste the gene list into <em>Query</em> and</li>
<li>in Options set Organism to Home sapiens and Target to Mus musculus.</li>
<li>Click on the orange button <em>Run query</em>.</li>
</ul></li>
</ul>
<p align="center">
<p><img src="Module5/chipseqlab/chipseqlab_image/gorth1.png" alt="gorth1" width="75%" /></p>
<ul>
<li>Click on the icon next to the ortholog name column to copy the gene list. This is the gene list containing the mouse orthologs of the RUNX1 targets that we used in Exercise 4.</li>
</ul>
<div class="float">
<img src="Module5/chipseqlab/chipseqlab_image/gorth2.png" alt="gorth2" />
<div class="figcaption">gorth2</div>
</div>
<p><strong>As reference (you dont need to go through these steps during the practical lab): Analysis of the RUNX1 Affy transcriptomics using GEO2R.</strong></p>
<ul>
<li>Go to the GEO page corresponding to the Affymetrix transcriptomics <a href="data:https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE129038" class="uri">data:https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE129038</a></li>
<li>Click on Analyze with GEO2R</li>
<li>Define your groups and click on Analyze</li>
<li>Export the table</li>
<li>Rank the genes using the absolute value of t</li>
<li>Remove the gene name duplicates</li>
<li>Select the top 500 genes up regulated using the largest t value and the 500 genes down regulated using the smallest t value</li>
</ul>
</div>
<div id="end-of-lab" class="section level2 hasAnchor">
<h2>End of Lab<a href="module-8.html#end-of-lab" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Congratulations!!</p>
</div>
<div id="optional-module-8-lab-2-gene-regulation-and-motif-analysis-practical-lab-iregulon" class="section level2 hasAnchor">
<h2>Optional Module 8 Lab 2: Gene Regulation and Motif Analysis Practical Lab / iRegulon<a href="module-8.html#optional-module-8-lab-2-gene-regulation-and-motif-analysis-practical-lab-iregulon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="goal-3" class="section level2 hasAnchor">
<h2>Goal<a href="module-8.html#goal-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Import a Cytoscape network and apply iRegulon on all the selected nodes.</li>
<li>Explore and understand the main output features of iRegulon such as the Transcription target view.</li>
<li>Learn how to display predicted targets of a specific transcription factor by creating its metatargetome.</li>
</ul>
<p>This practical consists of 2 exercises. Follow the step-by-step checklist through the exercises. Some notes about iRegulon and information about the output values are written at the end of the document.</p>
<p>Before starting the exercises, download the files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select Save Link As.
</p>
<p>
Place it in the corresponding module directory of your CBW work
directory.
</p>
</div>
<ul>
<li><a href="./Module5/iregulon/data/prostate_cancer_genemania_network.txt">prostate_cancer_genemania_network.txt</a></li>
</ul>
<div class="rmd-note">
<p>
in case the iRegulon server is not working, it is possible to work
with pre-computed results. Please look at the instructions at the bottom
of this page.
</p>
</div>
</div>
<div id="exercise-1.-detect-regulons-from-co-expressed-genes" class="section level2 hasAnchor">
<h2>Exercise 1. Detect regulons from co-expressed genes<a href="module-8.html#exercise-1.-detect-regulons-from-co-expressed-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this exercise, we are using genes frequently mutated in prostate cancer. iRegulon requires a network in order to start. We will use a GeneMANIA network that we previously saved for this purpose. Using iRegulon, we will look for transcription factors (TFs) that may regulate a set of genes in this network.
Note: iRegulon also accepts a simple gene list as input to create the network</p>
<p>To start this exercise, download to your computer the <a href="./Module5/iregulon/data/prostate_cancer_genemania_network.txt">prostate_cancer_genemania_network.txt</a> file.</p>
<div id="skills-learned-in-this-exercise" class="section level3 hasAnchor">
<h3>Skills learned in this exercise:<a href="module-8.html#skills-learned-in-this-exercise" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create a network by importing a text file, run iRegulon to detect regulons, explore the iRegulon results, create a regulon subnetwork, save the results.</p>
</div>
<div id="steps" class="section level3 hasAnchor">
<h3>Steps<a href="module-8.html#steps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Launch Cytoscape. Close the Welcome to Cytoscape window, if its enabled.</li>
</ol>
<p>Double click on the <img src="Module5/iregulon/images/cytoscape.png" alt="Cytoscape icon" />. Cytoscape icon.</p>
<ol start="2" style="list-style-type: decimal">
<li>Create a network using the prostate_cancer_genemania_network.txt file.
<ul>
<li><p>In the menu bar select File &gt; Import &gt; Network from File. A file open dialog pops up.<br><img src="Module5/iregulon/images/gp1_2a.png" alt="gp1_2a" /></p></li>
<li><p>Browse and locate the prostate_cancer_genemania_network.txt file. Click the Open button. An Import Network From Table dialog pops up.<br><img src="Module5/iregulon/images/2b.png" alt="2b" /></p></li>
<li><p>Select the column Entity 1 .</p></li>
<li><p>Expand the menu using the arrow on the right and click the green circle button to set this column as Source Node. <br><img src="Module5/iregulon/images/2c.png" alt="2c" /></p></li>
<li><p>Select the column Entity 2.</p></li>
<li><p>Click the red bullseye to set this column as Target Node.<br><img src="Module5/iregulon/images/2d.png" alt="2d" /></p></li>
<li><p>Click the OK button.</p></li>
</ul></li>
</ol>
<p>The main window now displays the created network. Each node represents a gene. Edges represent the relationships (e.g physical interactions, co-expression) between the genes (nodes) that were calculated by GeneMANIA.</p>
<div class="float">
<img src="Module5/iregulon/images/2e.png" alt="2e" />
<div class="figcaption">2e</div>
</div>
<div class="rmd-tip">
<p>
The shortcut +L (Mac) or Ctrl+L (Windows) is a quicker way to import
a network from a file.
</p>
</div>
<div class="rmd-tip">
<p>
If you only see gray nodes, go to Style and choose default style.
</p>
</div>
<ol start="3" style="list-style-type: decimal">
<li>Improve the layout.
<ul>
<li>In the menu bar, select Layout &gt; yFiles Organic Layout ( you need to install the yFiles layout algorithms app in Cytoscape app manager)<br><img src="Module5/iregulon/images/gp1_3a.png" alt="gp1_3a" /> <br><img src="Module5/iregulon/images/gp1_3b.png" alt="gp1_3b" /></li>
</ul></li>
<li>Select all nodes in the network. Click anywhere on the blank space and press +A (Mac) or Ctrl+A (Windows). Or you can also do this using the mouse. Click shift and drag from an empty space to the left of and above every node to an empty space to the right of and beneath every node. The selected nodes are now colored yellow.</li>
</ol>
<p><img src="Module5/iregulon/images/gp1_4.png" alt="workflow" width="60%" /></p>
<ol start="5" style="list-style-type: decimal">
<li>In the menu bar, select Apps &gt; iRegulon &gt; Predict regulators and targets.A Predict regulators and targets dialog pops up.
<ul>
<li>Using the default parameters, click the Submit button at the bottom of the page.A progress bar will pop up.</li>
<li>Wait until the running analysis is completed (usually less than 1 min). The progress bar will vanish, and a new right panel, Results Panel will be added to the main Cytoscape window.</li>
<li>Deselect all nodes by clicking on a blank space of the screen. The nodes are all cyan again.</li>
</ul></li>
</ol>
<div class="float">
<img src="Module5/iregulon/images/5a.png" alt="5a" />
<div class="figcaption">5a</div>
</div>
<div class="float">
<img src="Module5/iregulon/images/5b.png" alt="5b" />
<div class="figcaption">5b</div>
</div>
<ol start="6" style="list-style-type: decimal">
<li>Explore the results.
<ul>
<li>Locate the Results Panel on the right side of the window.</li>
<li>Click on the float window icon located at the upper right corner.</li>
</ul></li>
</ol>
<div class="rmd-tip">
<p>
resize the Result Panel window by expanding it horizontally and
vertically, so you can see the results and the network
simultaneously.
</p>
</div>
<div class="rmd-tip">
<p>
mouse over column names to get a tooltip describing their meaning in
more detail.
</p>
</div>
<div class="float">
<img src="Module5/iregulon/images/6.png" alt="6" />
<div class="figcaption">6</div>
</div>
<ol start="7" style="list-style-type: decimal">
<li>Explore the enrichment results in the Motifs tab from the Results Panel. It is a list of all DNA binding motifs that appear in more than one gene region from the prostate cancer gene list. They are ranked by the strongest Normalized Enrichment Score (NES). Some DNA binding motifs in the databases are related to a specific transcription factor, but others are not.
<ul>
<li>Check that Motifs is the selected tab of the Results Panel.</li>
<li>Click on the row for this motif to display the motifs sequence logo and related information at the bottom part of Results Panel.</li>
</ul></li>
</ol>
<p><img src="Module5/iregulon/images/7.png" alt="7" />
On the above screenshot, there is an enrichment in the prostate gene list for a motif called
YOL108C from the yetfasco database. The motif logo is displayed and it is very similar to the MITF binding motif. The genes from our network carrying this motif in their promoter region are indicated in red (TargetName). The rank indicates the number of motifs that they carry in their promoter region.</p>
<div class="rmd-tip">
<p>
Additional explanation about the results are located at the end of
this document and in more detail in the iRegulon reference paper.
</p>
</div>
<ol start="8" style="list-style-type: decimal">
<li>Explore the enrichment results in the Tracks tab. It is a list of all ChIP-seq datasets (or tracks) sorted by strongest enrichment from genes inour network.
<ul>
<li>Select the Tracks tab of the Results Panel.</li>
<li>Find a ClusterCode assigned to more than one track.</li>
</ul></li>
</ol>
<p><img src="Module5/iregulon/images/8.png" alt="8" />
T4 is a track cluster associated with 2 tracks and is highlighted in the table as an example.The 2 tracks are biological replicates (Rep1, Rep2) of a same chIP-seq experiment. The transcription factor used for this chIP_seq experiment is TCF12. The first track is ranked number 4 and the second track is ranked number 8. The genes with TCF12 peaks in their promoter regions are listed in red under TargetName.</p>
<ol start="9" style="list-style-type: decimal">
<li>Explore the enrichment results in the Transcription Factors tabview. This is the most important tab as each row is a transcription factor that is a potential co-regulator of the genes in our network. Each row represents a cluster that combines the results of the related motifs (Motifs tab) or tracks (Tracks tab) or both.
<ul>
<li>Select the Transcription Factors tab of the Results Panels.</li>
<li>Click on MTF1 and explore the results.</li>
</ul></li>
</ol>
<div class="float">
<img src="Module5/iregulon/images/9.png" alt="9" />
<div class="figcaption">9</div>
</div>
<p>MTF1 is associated with the motif cluster M1. This cluster contains 6 related motifs and 11 potential target genes. One motif (homer-M00129) selected as example in the above screenshot is directly annotated to the TFs NRF1 and ZSCAN10 as indicated by green checked signs.</p>
<ol start="10" style="list-style-type: decimal">
<li>How did iRegulon perform? Is MTF1 (metal-transcription factor 1) known to be expressed or to play a role in prostate cancer?</li>
</ol>
<div class="rmd-tip">
<p>
Open your web browser and search the web for [MTF1 prostate
cancer].
</p>
</div>
<div class="float">
<img src="Module5/iregulon/images/10.png" alt="10" />
<div class="figcaption">10</div>
</div>
<p>This network highlights MTF1 and interactions with other genes and miRs. This is a network involved in prostate cancer.<br>
<a href="PMID:14568174" class="uri">PMID:14568174</a><br>
<a href="PMID:23157640" class="uri">PMID:23157640</a></p>
<ol start="11" style="list-style-type: decimal">
<li>Add MTF1 to the network.
<ul>
<li>Check that the Transcription Factors tab is selected.</li>
<li>Click the MTF1 row to select it.</li>
<li>Click the Add regulator icon <img src="Module5/iregulon/images/add.png" alt="Add icon" /> located at the upper left corner of the Results Panel.
This adds MTF1 to the network as a yellow node, with the edges linking to its 11 potential targets, all highlighted as purple nodes.</li>
</ul></li>
</ol>
<p>11a)</p>
<div class="float">
<img src="Module5/iregulon/images/11a.png" alt="11a" />
<div class="figcaption">11a</div>
</div>
<p>11b)</p>
<p><img src="Module5/iregulon/images/gp1_11b.png" alt="workflow" width="60%" /></p>
<ol start="12" style="list-style-type: decimal">
<li>Create a subnetwork to better visualize the predicted targets.
<ul>
<li>Select the MTF1 node in the network by clicking on it.</li>
<li>In the Cytoscape toolbar above the network, click the First Neighbors of Selected Nodes icon <img src="Module5/iregulon/images/gp1_neighbours.png" alt="gp1_neighbours.png" />. MTF1 and its targets are now highlighted in yellow (which means they are selected).</li>
<li>Use the New network from selection icon <img src="Module5/iregulon/images/new.png" alt="New icon" /> to create a subnetwork.</li>
</ul></li>
</ol>
<p>12a)</p>
<p><img src="Module5/iregulon/images/gp1_12a.png" alt="workflow" width="60%" /></p>
<p>12b)</p>
<div class="float">
<img src="Module5/iregulon/images/12b.png" alt="12b" />
<div class="figcaption">12b</div>
</div>
<div class="rmd-tip">
<p>
If the node colors are not purple, go to Style and choose iRegulon
Visual Style.
</p>
</div>
<div class="float">
<img src="Module5/iregulon/images/gp1_12c.png" alt="gp1_12c" />
<div class="figcaption">gp1_12c</div>
</div>
<ol start="13" style="list-style-type: decimal">
<li>Add to the figure information on the types of interactions obtained from GeneMANIA and stored as additional information in the prostate_cancer_genemania_network.txt file.
<ul>
<li>In the Control Panel at the left of the window, select the Style tab. At the bottom of the panel, select the Edge tab.</li>
<li>Locate the Stroke Color property and click the right triangle to expand the box.</li>
<li>Change the Column field to Network group</li>
<li>Verify that the Mapping Type field is Discrete Mapping</li>
<li>For the first interaction type, choose a color by clicking on the Edit color button on the right side of the color field. Choose a color and click the OK button.</li>
<li>Repeat that step, choosing a different color for each interaction type.
The edges should now be colored by the types of interactions.</li>
</ul></li>
</ol>
<p>13a)</p>
<div class="float">
<img src="Module5/iregulon/images/gp1_13a.png" alt="gp1_13a" />
<div class="figcaption">gp1_13a</div>
</div>
<p>13b)</p>
<div class="float">
<img src="Module5/iregulon/images/gp1_13b.png" alt="gp1_13b" />
<div class="figcaption">gp1_13b</div>
</div>
<ol start="14" style="list-style-type: decimal">
<li>Save current results as an iRegulon (iRF) file.
<ul>
<li>In the Results Panel toolbar, click the Save current results as an iRegulon (iRF) file button <img src="Module5/iregulon/images/save1.png" alt="Save icon" />..</li>
<li>Choose a name and click the Save button.</li>
</ul></li>
</ol>
<div class="rmd-tip">
<p>
you can reuse these iRegulon results by loading this iRF file using
the Load saved results icon <img
src="Module5/iregulon/images/save2.png" alt="Save2 icon" />..
</p>
</div>
<p>14a)</p>
<div class="float">
<img src="Module5/iregulon/images/14.png" alt="14" />
<div class="figcaption">14</div>
</div>
<ol start="15" style="list-style-type: decimal">
<li>Save the Cytoscape session .
<ul>
<li>In the Cytoscape menu bar, select File &gt; Save as.</li>
<li>Choose a name and click the Save button.</li>
</ul></li>
</ol>
<div class="rmd-tip">
<p>
you can re-open this file later to examine the network further.
</p>
</div>
<div class="float">
<img src="Module5/iregulon/images/15.png" alt="15" />
<div class="figcaption">15</div>
</div>
</div>
</div>
<div id="exercise-2.-create-a-metatargetome-using-iregulon-and-merge-2-networks-in-cytoscape." class="section level2 hasAnchor">
<h2>Exercise 2. Create a metatargetome using iRegulon and merge 2 networks in Cytoscape.<a href="module-8.html#exercise-2.-create-a-metatargetome-using-iregulon-and-merge-2-networks-in-cytoscape." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This exercise does not require additional files.</p>
<p>This exercise will teach you to use the metatargetome function of iRegulon. This function displays a list of potential targets for a specific TF. We will create the metatargetome of two TFs, that we found as potential coregulators of the prostate cancer genes (exercise 1): MTF1 and LARP4. We will then learn how to use Cytoscape to merge two networks and visualize nodes in common.</p>
<p><strong>Steps</strong></p>
<ol style="list-style-type: decimal">
<li>Launch Cytoscape.
<ul>
<li>If Cytoscape is already opened, do File &gt; New &gt; Session. A Current session will be lost. Do you want to continue? dialog opens. Click on OK.<br />
</li>
<li>Double click on the Cytoscape icon.</li>
</ul></li>
<li>Create the metatargetome for MTF1.
<ul>
<li>From the menu bar , select File &gt; Apps &gt; iRegulon&gt; Query TF-target database.A Query TF-target database for a factor window pops up.</li>
<li>In the Transcription Factor field, select MTF1.</li>
<li>Set Network &gt; Number nodes (approx.) to 100.</li>
<li>Click the Submit button.</li>
</ul></li>
</ol>
<p>2a)</p>
<div class="float">
<img src="Module5/iregulon/images/2a2.png" alt="2a2" />
<div class="figcaption">2a2</div>
</div>
<p>2b)</p>
<div class="float">
<img src="Module5/iregulon/images/2b2.png" alt="2b2" />
<div class="figcaption">2b2</div>
</div>
<p>2c)</p>
<div class="float">
<img src="Module5/iregulon/images/2c2.png" alt="2c2" />
<div class="figcaption">2c2</div>
</div>
<ol start="3" style="list-style-type: decimal">
<li>Create the metatargetome for LARP4. Follow same steps as above.
<ul>
<li>From the Cytoscape menu bar, select File &gt; Apps&gt;iRegulon&gt; Query TF-target database.</li>
<li>A Query TF-target database for a factor window pops up. In the Transcription Factor field, enter LARP4.</li>
<li>Set Network &gt; Number nodes (approx.) to 100.</li>
<li>Click the Submit button.</li>
</ul></li>
</ol>
<p>3a)</p>
<div class="float">
<img src="Module5/iregulon/images/3a2.png" alt="3a2" />
<div class="figcaption">3a2</div>
</div>
<p>3b)</p>
<div class="float">
<img src="Module5/iregulon/images/3b2.png" alt="3b2" />
<div class="figcaption">3b2</div>
</div>
<ol start="4" style="list-style-type: decimal">
<li>Merge the two networks to visualize their shared target genes.
From the Cytoscape menu bar, select Tools &gt; Merge &gt; Networks.An Advanced Network Merge window pops up.
<ul>
<li>Check that the Union option is selected.</li>
<li>In the Available Networks list, select Metatargetome for LARP4.</li>
<li>Hold down the shift key while selecting Metatargetome for MTF1 so both networks are selected.</li>
<li>Click the right arrow to move the networks to the Networks to Merge list.</li>
<li>Click the Merge button.
Cytoscape now displays the two networks in the same window, linked by the two genes they have in common.</li>
</ul></li>
</ol>
<p>4a)</p>
<div class="float">
<img src="Module5/iregulon/images/4a2.png" alt="4a2" />
<div class="figcaption">4a2</div>
</div>
<p>4b)</p>
<div class="float">
<img src="Module5/iregulon/images/4b2.png" alt="4b2" />
<div class="figcaption">4b2</div>
</div>
<p>4c)</p>
<div class="float">
<img src="Module5/iregulon/images/4c2.png" alt="4c2" />
<div class="figcaption">4c2</div>
</div>
<div id="end-of-exercise" class="section level4 hasAnchor">
<h4>END OF EXERCISE<a href="module-8.html#end-of-exercise" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="use-our-precomputed-iregulon-results" class="section level3 hasAnchor">
<h3>Use our precomputed iRegulon results:<a href="module-8.html#use-our-precomputed-iregulon-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Download these files on your computer:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select Save Link As.
</p>
<p>
Place it in the corresponding module directory of your CBW work
directory.
</p>
</div>
<ul>
<li><p><a href="./Module5/iregulon/data/prostate_cancer_genemania_network.cys">prostate_cancer_genemania_network.cys</a></p></li>
<li><p><a href="./Module5/iregulon/data/iregulon_results.irf">iregulon_results.irf</a></p></li>
</ul>
<ol style="list-style-type: decimal">
<li><p>launch Cytoscape</p></li>
<li><p>open the prostate_cancer_genemania_network.cys file</p></li>
<li><p>go to App &gt; iRegulon &gt; Load results from the iregulon_results.irf file</p></li>
</ol>
</div>
<div id="notes-about-iregulon" class="section level3 hasAnchor">
<h3>Notes about iRegulon:<a href="module-8.html#notes-about-iregulon" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Website: <a href="http://iregulon.aertslab.org/" class="uri">http://iregulon.aertslab.org/</a>
Tutorials: <a href="http://iregulon.aertslab.org/tutorial.html#clusters" class="uri">http://iregulon.aertslab.org/tutorial.html#clusters</a>
Paper: <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003731" class="uri">http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003731</a> [<a href="PMID:25058159" class="uri">PMID:25058159</a>]</p>
<div id="motif-oriented-view" class="section level4 hasAnchor">
<h4>Motif oriented view:<a href="module-8.html#motif-oriented-view" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Each line is a DNA binding motif those sequence has been located in 20 kb regions centered around the TSS (transcription start site) of genes from the prostate cancer list (= genes in the network). The genes from the network which contained this DNA binding motif are called the target genes and displayed in the Target Name column. Their ranks are also indicated.</p>
<p>DNA binding motifs more usually represent a family of transcription factors (e.g.helix loop helix TFs ) rather than being specific to one particular TF. In addition, related TFs (e.g GATA1, GATA2, GATA3) can bind to very similar DNA sequences. iRegulon uses the motif2TF algorithm to associate a motif with a specific TF. The #TF column indicates which motifs are significantly associated to a TF (# &gt;= 1) or not (# = 0). Clicking on a motif line will display a panel indicating several related information. It will display all the TFs found significantly associated with the motif.</p>
<p>How is the enrichment calculated? (NES AUC) motif detection and enrichment score in a set of input genes.
iRegulon uses precomputed results to calculate for each motif the AUC (Area Under the cumulative Recovery Curve) and the NES (Normalized Enrichment Score). iRegulon accesses this database of precomputed results using a server connection when a search is launched.</p>
<p><strong>What are these precomputed results :</strong></p>
<p>iRegulon gathered known DNA binding motifs and their corresponding PWM (position weight matrix, see lecture) from different databases (eg TRANSFAC pro) (9713 PWMs). They then ranked all genes in the genome (22284 genes) for each motif from the most likely target of this motif to the least one (available for Human, Mouse and Drosophila).</p>
<p><strong>Calculating enrichment for our set of genes (our network) :</strong></p>
<p>Each ranked list (each motif) is then tested with our set of genes to see whether genes in our list are located more at the top of the ranked list (most likely targets of the motifs). From this ranked list and the overlap with our gene list, the AUC (Area Under the cumulative Recovery Curve) is calculated for each motif. The AUC is going to be larger if we have more genes at the top of our list. The higher the AUC values and the higher the tested motif is likely to co-regulate our genes (or some of them). The NES is derived from the AUC. The optimal subset of highly ranked lists are set as the potential target genes and displayed in the target name column.</p>
<p><strong>How are several motifs being similarly grouped under a same cluster code?</strong></p>
<p>To find TF associated with motifs, iRegulon uses the motif2TF algorithm. During this computation of motif2TF, motifs sharing similarities are grouped together and form a cluster. Within this cluster, some motifs are already known to correspond to a specific TF (direct annotation). This information is used to associate a motif with one or more related TFs. The ClusterCode column indicates the cluster assigned to each motif.</p>
<p><strong>Tracks oriented view:</strong></p>
<p>Each line is an ENCODE Chip_Seq track. Chip_seq are sequencing of fragments bound to a specific TF after immunoprecipitation of the TF and the DNA fragments. Each track is then specific to a transcription factor (the #TFs columns is always equal to 1). Clusters contain more than one track only if these tracks were generated using the same TF. All the values (NES, AUC, are the same for the motif, track of transcription factor oriented views.</p>
<p><strong>Transcription Factors oriented view:</strong></p>
<p>Each line is a cluster of motifs and or tracks and as the next column (TF) the best representative TF of this cluster determined by the motif2TF algorithm. All the values (NES, AUC, are the same for the motif, track of transcription factor oriented views.</p>
<p><strong>Metatargetome:</strong></p>
<p>iRegulon uses the pre-computed results not only for finding regulons but also for displaying the potential gene targets for any TF of interest available in the iRegulon database. Users can define the number of top potential targets they want to display. The result is visualized as a network using a circular layout with the TF of interest in the center of the network.</p>
</div>
</div>
<div id="notes-about-cytoscape" class="section level3 hasAnchor">
<h3>Notes about Cytoscape:<a href="module-8.html#notes-about-cytoscape" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Link to tutorials showing how to format data to create a Cytoscape network starting from a simple gene list:
<a href="http://wiki.cytoscape.org/Cytoscape_User_Manual/Network_Formats" class="uri">http://wiki.cytoscape.org/Cytoscape_User_Manual/Network_Formats</a></p>
<p><strong>Note about organic layout:</strong></p>
<p>The organic layout style is based on the force-directed layout paradigm. When calculating a layout, the nodes are considered to be physical objects with mutually repulsive forces, like, e.g., protons or electrons. The connections between nodes also follow the physical analogy and are considered to be springs attached to the pair of nodes.  The layout algorithm simulates these physical forces and rearranges the positions of the nodes in such a way that the sum of the forces emitted by the nodes and the edges reaches a (local) minimum.</p>
<p>Resulting layouts often expose the inherent symmetric and clustered structure of a graph, they show a well-balanced distribution of nodes and have few edge crossings. <a href="http://docs.yworks.com/yfiles/doc/developers-guide/smart_organic_layouter.html" class="uri">http://docs.yworks.com/yfiles/doc/developers-guide/smart_organic_layouter.html</a> .</p>
<div id="section-2" class="section level60">
<p class="heading"></p>
</div>
</div>
</div>
<div id="exercise-3.-use-enrichr-with-the-prostate-gene-list." class="section level2 hasAnchor">
<h2>Exercise 3. Use Enrichr with the prostate gene list.<a href="module-8.html#exercise-3.-use-enrichr-with-the-prostate-gene-list." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before starting the exercise, download the files:</p>
<ul>
<li><a href="./Module5/iregulon/data/prostate_genelist.csv">prostate_genelist.csv</a></li>
<li><a href="./Module5/iregulon/data/TRRUST_Transcription_Factors_2019_table.txt">TRRUST_Transcription_Factors_2019_table.txt</a></li>
<li><a href="./Module5/iregulon/data/TTRUST_rank.xlsx">TTRUST_rank.xlsx</a></li>
</ul>
<div id="goal-4" class="section level3 hasAnchor">
<h3>Goal<a href="module-8.html#goal-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Use Enrichr on the prostate gene list and explore which transcription factors were predicted to be regulator on the same gene list used for the iRegulon lab.</p></li>
<li><p>After exploring the Enrichr results, we are going to export it into Cytoscape/EnrichmentMap. This is another opportunity to learn how to create a network and modify its style.</p></li>
</ul>
</div>
<div id="steps-1" class="section level3 hasAnchor">
<h3>Steps<a href="module-8.html#steps-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Launch Enrichr on a web browser using this address: <a href="https://amp.pharm.mssm.edu/Enrichr/" class="uri">https://amp.pharm.mssm.edu/Enrichr/</a></p></li>
<li><p>In the input data window, copy and paste the genes from the <a href="./Module5/iregulon/data/prostate_genelist.csv">prostate gene list</a></p></li>
</ol>
<div class="float">
<img src="Module5/iregulon/images/enrichr1.png" alt="enrichr1.png" />
<div class="figcaption">enrichr1.png</div>
</div>
<ol start="3" style="list-style-type: decimal">
<li><p>Click on the Submit button</p></li>
<li><p>The results are now displayed. Check that the Transcription tab is the one selected.<br><img src="Module5/iregulon/images/enrichr2.png" alt="enrichr2.png" /></p>
<ul>
<li>Explore the results from the different gene-set libraries on your own (CHEA 2016, TRANSFAC and JASPAR PWMs, etc) .</li>
</ul></li>
<li><p>Then, click on the gene-set library called TRRUST Transcription Factors 2019</p>
<ul>
<li>TRRUST (<a href="https://www.grnpedia.org/trrust/" class="uri">https://www.grnpedia.org/trrust/</a>) is a manually curated database of human and mouse transcriptional regulatory networks. Each gene-set contained some target genes for a particular transcription factor. It contains mouse and human data. They have been derived from pubmed articles which describe small-scale experimental studies of transcriptional regulations.</li>
<li>We are going to explore the result in this library as some gene-sets are significantly enriched at FDR &lt; 0.05.<br><img src="Module5/iregulon/images/enrichr3.png" alt="enrichr3.png" /></li>
<li>The observation of the bar graph indicates that the transcription factor NR5A1 is the most significant result.</li>
</ul></li>
<li><p>Click on the Table to display the results as a table.<br><img src="Module5/iregulon/images/enrichr4.png" alt="enrichr4.png" /></p>
<ul>
<li>Remember from the presentation that the Adjusted p-value represents the FDR. As the FDR is less than 0.05, all these gene-sets are significantly enriched in our gene list.</li>
</ul></li>
<li><p>Click on the Export entries to table. Open the file that was downloaded on your computer in excel.<br><img src="Module5/iregulon/images/enrichr5.png" alt="enrichr5.png" /></p>
<ul>
<li>This table contains all the gene-sets significantly enriched or not.</li>
<li>The Term column contains the name of the transcription factors and the last column Genes contains the list of genes that are the targets of these transcription factors. All these genes are the ones present in the prostate gene list. The overlap 8/22 means that 22 genes are the known target of NR5A1 and 8 are present in the prostate gene list.</li>
<li>We are going to use this table to create an enrichment map in Cytoscape.</li>
</ul></li>
<li><p>Open Cytoscape.</p></li>
<li><p>Click in the menu bar on Apps and EnrichmentMap. A Create Enrichment Map dialog box opens.</p></li>
<li><p>Drag and drop the <a href="./Module5/iregulon/data/TRRUST_Transcription_Factors_2019_table.txt">TRRUST_Transcription_Factors_2019_table.txt</a> in the Data Sets window.</p>
<ul>
<li>On the right, check that the Analysis Type is set to Generic/gProfiler/Enrichr.</li>
<li>Set the FDR q-value cutoff at 0.05.</li>
</ul></li>
</ol>
<div class="float">
<img src="Module5/iregulon/images/enrichr6.png" alt="enrichr6.png" />
<div class="figcaption">enrichr6.png</div>
</div>
<ol start="10" style="list-style-type: decimal">
<li><p>Click on the Build button.</p></li>
<li><p>An enrichment map is now created.<br> <img src="Module5/iregulon/images/enrichr7.png" alt="workflow" width="60%" /></p>
<ul>
<li>The nodes are the transcription factor gene-sets. You can click on a node to see the genes that are the targets of these transcription factors. Transcription factors are connected by edges if they have target genes in common.</li>
</ul></li>
<li><p>Modify the visual style</p>
<ul>
<li>In the EnrichmentMap tab on the left, locate Style and set Chart Data to None.
<img src="Module5/iregulon/images/enrichr8.png" alt="enrichr8.png" /></li>
</ul></li>
<li><p>Import a file</p>
<ul>
<li>Our goal is to adjust node size and node color relatively to the gene-set enrichment results. To make it easier, a file has been created for you that ranks the gene-sets from 1 to 98 in the order of the adjusted p values. We will import this file in Cytoscape as a node table.</li>
<li>To import the file, locate File in the Cytoscape menu bar and then Import &gt; Table from File.<br><img src="Module5/iregulon/images/enrichr9.png" alt="enrichr9.png" /></li>
<li>Browse your computer to find the file <a href="./Module5/iregulon/data/TTRUST_rank.xlsx">TTRUST_rank.xlsx</a> that you have downloaded at the beginning of part 3 and click Open.</li>
<li>An Import Columns From Table dialog box opens. Click on OK.</li>
</ul></li>
</ol>
<div class="float">
<img src="Module5/iregulon/images/enrichr11.png" alt="enrichr11.png" />
<div class="figcaption">enrichr11.png</div>
</div>
<ol start="14" style="list-style-type: decimal">
<li>Play with the visual style
<ul>
<li>Locate the Cytoscape Style tab<br><img src="Module5/iregulon/images/enrichr10.png" alt="enrichr10.png" /></li>
<li>Locate the Cytoscape Style tab Fill Color property in the Node tab and expand the tab using the arrow on the right</li>
<li>Remove the current mapping using the trash can icon.<br><img src="Module5/iregulon/images/enrichr12.png" alt="enrichr12.png" /></li>
<li>In Column, choose myrank and in Mapping Type, choose Continuous Mapping.<br><img src="Module5/iregulon/images/enrichr13.png" alt="enrichr13.png" /></li>
<li>Locate the Size property and expand the tab using the arrow on the right</li>
<li>Remove the current mapping using the trash can icon.<br><img src="Module5/iregulon/images/enrichr14.png" alt="enrichr14.png" /></li>
<li>In Column, choose myrank and in Mapping Type, choose Continuous Mapping.</li>
<li>Set high node size values for low rank and low node size for high rank<br><img src="Module5/iregulon/images/enrichr15.png" alt="enrichr15.png" /></li>
<li>The enrichment map shows now in yellow and large nodes the transcription factors that were the most significantly enriched (based on the adjusted p value ranking). It also shows the links to the other gene-sets.<br><img src="Module5/iregulon/images/enrichr16.png" alt="enrichr16.png" /></li>
<li>NR5A1 (the most significant gene-set) is indeed known to be associated with prostate cancer. The prostate is a hormone-dependent organ. NR5A1 is a steroid nuclear receptor and has now been reported to be expressed in aggressive forms of prostate cancer (<a href="https://academic.oup.com/endo/article/155/2/358/2423115" class="uri">https://academic.oup.com/endo/article/155/2/358/2423115</a>).</li>
</ul></li>
</ol>
</div>
<div id="end-of-practical-lab" class="section level3 hasAnchor">
<h3>end of practical lab<a href="module-8.html#end-of-practical-lab" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Congratulations!</p>
</div>
</div>
<div id="optional-module-8-lab-3-automated-enrichment-and-visualisation-lab-using-clusterprofiler" class="section level2 hasAnchor">
<h2>Optional Module 8 Lab 3: Automated Enrichment and Visualisation Lab using <code>clusterProfiler</code><a href="module-8.html#optional-module-8-lab-3-automated-enrichment-and-visualisation-lab-using-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="clusterprofiler-lab" class="section level2 hasAnchor">
<h2><code>clusterProfiler</code> lab<a href="module-8.html#clusterprofiler-lab" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>clusterProfiler</code> is an R package that implements methods to perform both functional annotation and visualization of genes and gene clusters.</p>
<ul>
<li>It can accept data from a variety of experimental sources such as DNA-seq, RNA-seq, microarray, Mass spectometry, meRIP-seq, m6A-seq, ATAC-seq and ChIP-seq and thus can be applied in diverse scenarios.</li>
<li>It provides a tidy interface to access, manipulate, and visualize enrichment results to help users achieve efficient data interpretation.</li>
</ul>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a> is released within the <a href="https://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">Bioconductor project</a> and the source code is hosted on <a href="https://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">GitHub.</a></p>
</div>
<div id="goal-5" class="section level2 hasAnchor">
<h2>Goal<a href="module-8.html#goal-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Learn how to write R scripts for going from gene list to enriched pathways</li>
<li>Learn how to run over representation analysis (ORA) and gene set enrichment analysis (GSEA) using functions in the <a href="https://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a> R package</li>
<li>Explore results of enrichment analysis using various visualisation options in clusterProfiler</li>
</ul>
</div>
<div id="supported-analysis" class="section level2 hasAnchor">
<h2>Supported Analysis<a href="module-8.html#supported-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For functional annotation, <code>clusterprofiler</code> provides R functions to perform</p>
<ul>
<li>Over Representation Analysis</li>
<li>Gene Set Enrichment Analysis</li>
<li>Biological theme comparison</li>
</ul>
<p>In this practical, we will be learning how to run Over Representation Analysis and Gene Set Enrichment Analysis in 2 exercises. Follow the step-by-step checklist.</p>
<p>Before starting the exercises, make sure that <code>clusterProfiler</code> and other required packages are installed and loaded. Run prework_module8_clusterprofiler.R
before following this module.</p>
</div>
<div id="install-and-load-packages" class="section level2 hasAnchor">
<h2>Install and load packages<a href="module-8.html#install-and-load-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To run enrichment analysis using <code>clusterProfiler</code>, we need a few additional packages <code>org.Hs.eg.db</code>, <code>DOSE</code>, <code>tidyverse</code>, <code>enrichplot</code>, <code>ggupset</code>. Install and load all necessary packages using this code:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="module-8.html#cb19-1" tabindex="-1"></a><span class="co"># install and load the package  manager</span></span>
<span id="cb19-2"><a href="module-8.html#cb19-2" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb19-3"><a href="module-8.html#cb19-3" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb19-4"><a href="module-8.html#cb19-4" tabindex="-1"></a></span>
<span id="cb19-5"><a href="module-8.html#cb19-5" tabindex="-1"></a><span class="co"># list the required bioconductor packages </span></span>
<span id="cb19-6"><a href="module-8.html#cb19-6" tabindex="-1"></a>bio.pkgs <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;clusterProfiler&quot;</span>, <span class="st">&quot;org.Hs.eg.db&quot;</span>, <span class="st">&quot;DOSE&quot;</span>, <span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;enrichplot&quot;</span>, <span class="st">&quot;ggupset&quot;</span>)</span>
<span id="cb19-7"><a href="module-8.html#cb19-7" tabindex="-1"></a></span>
<span id="cb19-8"><a href="module-8.html#cb19-8" tabindex="-1"></a><span class="co"># install</span></span>
<span id="cb19-9"><a href="module-8.html#cb19-9" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(bio.pkgs)</span></code></pre></div>
<pre><code>## &#39;getOption(&quot;repos&quot;)&#39; replaces Bioconductor standard repositories, see
## &#39;help(&quot;repositories&quot;, package = &quot;BiocManager&quot;)&#39; for details.
## Replacement repositories:
##     CRAN: https://cran.rstudio.com/</code></pre>
<pre><code>## Bioconductor version 3.21 (BiocManager 1.30.26), R 4.5.0 (2025-04-11)</code></pre>
<pre><code>## Warning: package(s) not installed when version(s) same as or greater than current; use
##   `force = TRUE` to re-install: &#39;clusterProfiler&#39; &#39;org.Hs.eg.db&#39; &#39;DOSE&#39;
##   &#39;tidyverse&#39; &#39;enrichplot&#39; &#39;ggupset&#39;</code></pre>
<pre><code>## Old packages: &#39;BiocGenerics&#39;, &#39;bookdown&#39;, &#39;boot&#39;, &#39;broom&#39;, &#39;ClusterR&#39;,
##   &#39;cowplot&#39;, &#39;crosstalk&#39;, &#39;curl&#39;, &#39;data.table&#39;, &#39;dbplyr&#39;, &#39;dendextend&#39;, &#39;doBy&#39;,
##   &#39;DT&#39;, &#39;dtplyr&#39;, &#39;edgeR&#39;, &#39;emmeans&#39;, &#39;evaluate&#39;, &#39;FactoMineR&#39;, &#39;forcats&#39;,
##   &#39;gargle&#39;, &#39;GenomeInfoDb&#39;, &#39;ggplot2&#39;, &#39;ggpubr&#39;, &#39;ggsci&#39;, &#39;googledrive&#39;,
##   &#39;googlesheets4&#39;, &#39;here&#39;, &#39;hms&#39;, &#39;igraph&#39;, &#39;later&#39;, &#39;limma&#39;, &#39;magrittr&#39;,
##   &#39;Matrix&#39;, &#39;openssl&#39;, &#39;pbkrtest&#39;, &#39;pillar&#39;, &#39;promises&#39;, &#39;purrr&#39;, &#39;ragg&#39;,
##   &#39;Rcpp&#39;, &#39;RcppArmadillo&#39;, &#39;reformulas&#39;, &#39;reticulate&#39;, &#39;rgl&#39;, &#39;rmarkdown&#39;,
##   &#39;rprojroot&#39;, &#39;rstatix&#39;, &#39;rvest&#39;, &#39;SparseArray&#39;, &#39;statmod&#39;, &#39;stringr&#39;,
##   &#39;svglite&#39;, &#39;systemfonts&#39;, &#39;textshaping&#39;, &#39;vroom&#39;, &#39;xfun&#39;, &#39;xml2&#39;</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="module-8.html#cb24-1" tabindex="-1"></a><span class="co"># load all at once</span></span>
<span id="cb24-2"><a href="module-8.html#cb24-2" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(bio.pkgs, <span class="cf">function</span>(x) <span class="fu">library</span>(x, <span class="at">character.only=</span><span class="cn">TRUE</span>, <span class="at">quietly =</span> T)))</span></code></pre></div>
<pre><code>## </code></pre>
<pre><code>## clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/
## 
## Please cite:
## 
## Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.
## clusterProfiler: an R package for comparing biological themes among
## gene clusters. OMICS: A Journal of Integrative Biology. 2012,
## 16(5):284-287</code></pre>
<pre><code>## 
## Attaching package: &#39;clusterProfiler&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
<pre><code>## 
## Attaching package: &#39;generics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
##     setequal, union</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
##     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
##     unsplit, which.max, which.min</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:clusterProfiler&#39;:
## 
##     rename</code></pre>
<pre><code>## The following object is masked from &#39;package:utils&#39;:
## 
##     findMatches</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:clusterProfiler&#39;:
## 
##     slice</code></pre>
<pre><code>## 
## Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:clusterProfiler&#39;:
## 
##     select</code></pre>
<pre><code>## </code></pre>
<pre><code>## DOSE v4.2.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/
## 
## Please cite:
## 
## Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an
## R/Bioconductor package for Disease Ontology Semantic and Enrichment
## analysis. Bioinformatics. 2015, 31(4):608-609</code></pre>
<pre><code>##  Attaching core tidyverse packages  tidyverse 2.0.0 
##  dplyr     1.1.4      purrr     1.0.4
##  forcats   1.0.0      stringr   1.5.1
##  ggplot2   3.5.2      tibble    3.3.0
##  lubridate 1.9.4      tidyr     1.3.1
##  Conflicts  tidyverse_conflicts() 
##  lubridate::%within%() masks IRanges::%within%()
##  dplyr::collapse()     masks IRanges::collapse()
##  dplyr::combine()      masks Biobase::combine(), BiocGenerics::combine()
##  dplyr::desc()         masks IRanges::desc()
##  tidyr::expand()       masks S4Vectors::expand()
##  dplyr::filter()       masks clusterProfiler::filter(), stats::filter()
##  dplyr::first()        masks S4Vectors::first()
##  purrr::flatten()      masks jsonlite::flatten()
##  dplyr::lag()          masks stats::lag()
##  ggplot2::Position()   masks BiocGenerics::Position(), base::Position()
##  purrr::reduce()       masks IRanges::reduce()
##  dplyr::rename()       masks S4Vectors::rename(), clusterProfiler::rename()
##  lubridate::second()   masks S4Vectors::second()
##  lubridate::second&lt;-() masks S4Vectors::second&lt;-()
##  dplyr::select()       masks AnnotationDbi::select(), clusterProfiler::select()
##  purrr::simplify()     masks clusterProfiler::simplify()
##  dplyr::slice()        masks IRanges::slice(), clusterProfiler::slice()
##  Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre><code>## Warning: package &#39;enrichplot&#39; was built under R version 4.5.1</code></pre>
<pre><code>## enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/
## 
## Please cite:
## 
## Guangchuang Yu. Using meshes for MeSH term enrichment and semantic
## analyses. Bioinformatics. 2018, 34(21):3766-3767,
## doi:10.1093/bioinformatics/bty410</code></pre>
<p>Once all packages are loaded, we can get started with exercises.</p>
</div>
<div id="exercise-1a.-over-representation-analysis" class="section level2 hasAnchor">
<h2>Exercise 1a. Over representation analysis<a href="module-8.html#exercise-1a.-over-representation-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>clusterProfiler</code> supports over representation analysis against various sources such as GO annotation, KEGG pathway, MSigDB to name a few. For the full list please refer <a href="https://guangchuangyu.github.io/software/clusterProfiler/">this link.</a></p>
<p>In this exercise, we will learn over representation analysis using the gene ontology annotations. This is implemented in the function <code>enrichGO()</code>.</p>
<div id="data-for-enrichment-using-clusterprofiler" class="section level3 hasAnchor">
<h3>Data for enrichment using <code>clusterProfiler</code><a href="module-8.html#data-for-enrichment-using-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let us start with loading the dataset <code>geneList</code> that is provided by the package <code>DOSE</code>.</p>
<div class="rmd-note">
<p>
DOSE provides an example dataset <code>geneList</code>. It comes from
analysis of a <a
href="https://bioconductor.org/packages/release/data/experiment/html/breastCancerMAINZ.html">breast
cancer dataset</a> that had 200 samples, including 29 samples in grade
I, 136 samples in grade II and 35 samples in grade III. The ratios of
geometric means of grade III samples versus geometric means of grade I
samples were computed. Logarithm of these ratios (base 2) are stored in
<code>geneList</code> dataset.
</p>
</div>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="module-8.html#cb48-1" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span></code></pre></div>
<p>A variable called <code>geneList</code> should be loaded in your R environment. What does it look like?</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="module-8.html#cb49-1" tabindex="-1"></a><span class="fu">head</span>(geneList)</span></code></pre></div>
<pre><code>##     4312     8318    10874    55143    55388      991 
## 4.572613 4.514594 4.418218 4.144075 3.876258 3.677857</code></pre>
<div class="rmd-note">
<p>
As you can see, first line of output has names of genes in Entrez
gene ID format and the second line has fold change values of genes.
</p>
</div>
</div>
<div id="data-for-over-representation-analysis-using-clusterprofiler" class="section level3 hasAnchor">
<h3>Data for over representation analysis using <code>clusterProfiler</code><a href="module-8.html#data-for-over-representation-analysis-using-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For running an over representation analysis, we need only a list of gene names or IDs. Let us extract out the genes which had an expression value &gt;2 or &lt;-2 using the function <code>names()</code></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="module-8.html#cb51-1" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="dv">2</span>]</span>
<span id="cb51-2"><a href="module-8.html#cb51-2" tabindex="-1"></a></span>
<span id="cb51-3"><a href="module-8.html#cb51-3" tabindex="-1"></a><span class="fu">head</span>(gene)</span></code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
<p><code>gene</code> has a list of 207 genes.</p>
</div>
<div id="perform-go-over-representation-analysis" class="section level3 hasAnchor">
<h3>Perform GO over representation analysis<a href="module-8.html#perform-go-over-representation-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now, run <code>enrichGO()</code> with this list of genes and examine the results</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="module-8.html#cb53-1" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene          =</span> gene,</span>
<span id="cb53-2"><a href="module-8.html#cb53-2" tabindex="-1"></a>                <span class="at">universe      =</span> <span class="fu">names</span>(geneList),</span>
<span id="cb53-3"><a href="module-8.html#cb53-3" tabindex="-1"></a>                <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb53-4"><a href="module-8.html#cb53-4" tabindex="-1"></a>                <span class="at">ont           =</span> <span class="st">&quot;ALL&quot;</span>,</span>
<span id="cb53-5"><a href="module-8.html#cb53-5" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb53-6"><a href="module-8.html#cb53-6" tabindex="-1"></a>                <span class="at">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb53-7"><a href="module-8.html#cb53-7" tabindex="-1"></a>                <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb53-8"><a href="module-8.html#cb53-8" tabindex="-1"></a>                <span class="at">readable      =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="results-of-go-over-representation-analysis" class="section level3 hasAnchor">
<h3>Results of GO over representation analysis<a href="module-8.html#results-of-go-over-representation-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Examine the results. Do you notice any similarities or differences between this output format and your results from <strong>Module 2</strong> <code>gProfiler</code>?</p>
<p>The output table is stored in <code>ego@result</code>. In this example, 152 processes were significantly enriched.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="module-8.html#cb54-1" tabindex="-1"></a><span class="fu">head</span>(ego)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                          Description GeneRatio
## GO:0098813       BP GO:0098813       nuclear chromosome segregation    34/196
## GO:0000070       BP GO:0000070 mitotic sister chromatid segregation    28/196
## GO:0000819       BP GO:0000819         sister chromatid segregation    30/196
## GO:0140014       BP GO:0140014             mitotic nuclear division    32/196
## GO:0007059       BP GO:0007059               chromosome segregation    37/196
## GO:0000280       BP GO:0000280                     nuclear division    37/196
##              BgRatio RichFactor FoldEnrichment   zScore       pvalue
## GO:0098813 244/11536  0.1393443       8.201405 14.94710 8.216228e-22
## GO:0000070 158/11536  0.1772152      10.430380 15.69116 5.601659e-21
## GO:0000819 191/11536  0.1570681       9.244577 15.10474 7.595989e-21
## GO:0140014 230/11536  0.1391304       8.188820 14.47763 1.545602e-20
## GO:0007059 328/11536  0.1128049       6.639373 13.62178 1.843982e-20
## GO:0000280 331/11536  0.1117825       6.579197 13.53973 2.527225e-20
##                p.adjust       qvalue
## GO:0098813 2.446793e-18 2.263355e-18
## GO:0000070 7.540285e-18 6.974984e-18
## GO:0000819 7.540285e-18 6.974984e-18
## GO:0140014 1.098275e-17 1.015937e-17
## GO:0007059 1.098275e-17 1.015937e-17
## GO:0000280 1.254346e-17 1.160306e-17
##                                                                                                                                                                                                                                  geneID
## GO:0098813                     CDCA8/CDC20/KIF23/CENPE/MYBL2/CCNB2/NDC80/TOP2A/NCAPH/ASPM/DLGAP5/UBE2C/SKA1/NUSAP1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF18A/CDT1/BIRC5/KIF11/TTK/NCAPG/AURKB/TRIP13/PRC1/KIFC1/KIF18B/AURKA/CCNB1/KIF4A/PTTG1
## GO:0000070                                                        CDCA8/CDC20/KIF23/CENPE/MYBL2/NDC80/NCAPH/DLGAP5/UBE2C/SKA1/NUSAP1/TPX2/NEK2/CDK1/MAD2L1/KIF18A/CDT1/BIRC5/KIF11/TTK/NCAPG/AURKB/TRIP13/PRC1/KIFC1/KIF18B/CCNB1/KIF4A
## GO:0000819                                            CDCA8/CDC20/KIF23/CENPE/MYBL2/NDC80/TOP2A/NCAPH/DLGAP5/UBE2C/SKA1/NUSAP1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF18A/CDT1/BIRC5/KIF11/TTK/NCAPG/AURKB/TRIP13/PRC1/KIFC1/KIF18B/CCNB1/KIF4A
## GO:0140014                                 CDCA8/CDC20/KIF23/CENPE/MYBL2/NDC80/NCAPH/DLGAP5/UBE2C/SKA1/NUSAP1/TPX2/NEK2/UBE2S/CDK1/MAD2L1/KIF18A/CDT1/BIRC5/KIF11/TTK/NCAPG/AURKB/CHEK1/TRIP13/PRC1/KIFC1/KIF18B/AURKA/CCNB1/KIF4A/BMP4
## GO:0007059   CDCA8/CDC20/KIF23/CENPE/MYBL2/CCNB2/NDC80/TOP2A/NCAPH/ASPM/DLGAP5/UBE2C/HJURP/SKA1/NUSAP1/TPX2/TACC3/NEK2/CENPM/CENPN/CDK1/MAD2L1/KIF18A/CDT1/BIRC5/KIF11/TTK/NCAPG/AURKB/TRIP13/PRC1/KIFC1/KIF18B/AURKA/CCNB1/KIF4A/PTTG1
## GO:0000280 CDCA8/CDC20/KIF23/CENPE/MYBL2/CCNB2/NDC80/TOP2A/NCAPH/ASPM/DLGAP5/UBE2C/SKA1/NUSAP1/TPX2/NEK2/RAD51AP1/UBE2S/CDK1/MAD2L1/KIF18A/CDT1/BIRC5/KIF11/TTK/NCAPG/AURKB/CHEK1/TRIP13/PRC1/KIFC1/KIF18B/AURKA/CCNB1/KIF4A/PTTG1/BMP4
##            Count
## GO:0098813    34
## GO:0000070    28
## GO:0000819    30
## GO:0140014    32
## GO:0007059    37
## GO:0000280    37</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="module-8.html#cb56-1" tabindex="-1"></a><span class="fu">nrow</span>(ego<span class="sc">@</span>result)</span></code></pre></div>
<pre><code>## [1] 152</code></pre>
</div>
<div id="input-options-for-enrichgo" class="section level3 hasAnchor">
<h3>Input options for <code>enrichGO()</code>:<a href="module-8.html#input-options-for-enrichgo" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="rmd-note">
<ul>
<li>
The default option for <code>gene</code> is entrez gene ID, but
other gene ID formats are supported in GO analyses. You should specify
the <code>keyType</code> parameter to specify the input gene ID type
(More details below)
</li>
<li>
We have selected all genes measured in the experiment as our
<code>universe.</code>
</li>
<li>
You can specify subontology using the argument <code>ont</code>. It
takes one option among - BP, MF, CC or ALL for biological
process, molecular function, cellular co-localization or all
subontologies respectively.
</li>
<li>
If <code>readable</code> is set to <code>TRUE</code>, the input gene
IDs will be converted to gene symbols.
</li>
<li>
OrgDb is the genome annotation database of organism that your gene
list is coming from. Since our <code>geneList</code> is from human
breast cancer, we have provided human OrgDb object
(<code>org.Hs.eg.db</code>). See the section A note on supported
organisms for more details.
</li>
</ul>
<p>
Gene IDs can be converted to different formats using
<code>bitr()</code> function.
</p>
</div>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="module-8.html#cb58-1" tabindex="-1"></a><span class="co"># convert from entrez gene ID to ensembl ID and gene symbols</span></span>
<span id="cb58-2"><a href="module-8.html#cb58-2" tabindex="-1"></a>gene.df <span class="ot">&lt;-</span> <span class="fu">bitr</span>(gene, </span>
<span id="cb58-3"><a href="module-8.html#cb58-3" tabindex="-1"></a>                <span class="at">fromType =</span> <span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb58-4"><a href="module-8.html#cb58-4" tabindex="-1"></a>                <span class="at">toType =</span> <span class="fu">c</span>(<span class="st">&quot;ENSEMBL&quot;</span>, <span class="st">&quot;SYMBOL&quot;</span>),</span>
<span id="cb58-5"><a href="module-8.html#cb58-5" tabindex="-1"></a>                <span class="at">OrgDb =</span> org.Hs.eg.db)</span></code></pre></div>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in bitr(gene, fromType = &quot;ENTREZID&quot;, toType = c(&quot;ENSEMBL&quot;, &quot;SYMBOL&quot;), :
## 0.48% of input gene IDs are fail to map...</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="module-8.html#cb61-1" tabindex="-1"></a><span class="fu">head</span>(gene.df)</span></code></pre></div>
<pre><code>##   ENTREZID         ENSEMBL SYMBOL
## 1     4312 ENSG00000196611   MMP1
## 2     8318 ENSG00000093009  CDC45
## 3    10874 ENSG00000109255    NMU
## 4    55143 ENSG00000134690  CDCA8
## 5    55388 ENSG00000065328  MCM10
## 6      991 ENSG00000117399  CDC20</code></pre>
<div class="rmd-note">
<p>
Various options for <code>keyType</code> can be found using
<code>keytypes(&lt;name of organism annotation&gt;)</code>. For example:
<code>keytypes(org.Hs.eg.db)</code>
</p>
</div>
</div>
<div id="simplify-enrichgo-results" class="section level3 hasAnchor">
<h3>Simplify <code>enrichGO()</code> results<a href="module-8.html#simplify-enrichgo-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>GO enrichment typically contains redundant terms. You <strong>may</strong> use the <code>simplify()</code> function to reduce redundancy of enriched GO terms using the default parameters. Please note that simplifying is not always a necessary step. You can choose to omit it, based on the nature of your result tables.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="module-8.html#cb63-1" tabindex="-1"></a>ego.sim <span class="ot">=</span> clusterProfiler<span class="sc">::</span><span class="fu">simplify</span>(ego)</span>
<span id="cb63-2"><a href="module-8.html#cb63-2" tabindex="-1"></a><span class="fu">nrow</span>(ego.sim)</span></code></pre></div>
<pre><code>## [1] 43</code></pre>
</div>
</div>
<div id="exercise-1b.-visualise-the-results-of-go-over-representation-analysis" class="section level2 hasAnchor">
<h2>Exercise 1b. Visualise the results of GO over representation analysis<a href="module-8.html#exercise-1b.-visualise-the-results-of-go-over-representation-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="barplot" class="section level3 hasAnchor">
<h3>Barplot<a href="module-8.html#barplot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Bar plot is the most widely used method to visualize enriched terms. It shows the enrichment scores (e.g.p values) and gene count or ratio as bar height and color. You can specify the number of terms (most significant) to display via the <code>showCategory</code> parameter.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="module-8.html#cb65-1" tabindex="-1"></a><span class="fu">barplot</span>(ego.sim, <span class="at">showCategory=</span><span class="dv">20</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;ORA barplot (top 20)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-210-1.png" width="672" />
You can plot other variables such as <code>log10(p.adjust)</code> by modifying using <code>mutate()</code> from the <code>tidyverse</code> package</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="module-8.html#cb66-1" tabindex="-1"></a><span class="fu">mutate</span>(ego.sim, <span class="at">qscore =</span> <span class="sc">-</span><span class="fu">log</span>(p.adjust, <span class="at">base=</span><span class="dv">10</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb66-2"><a href="module-8.html#cb66-2" tabindex="-1"></a>    <span class="fu">barplot</span>(<span class="at">x=</span><span class="st">&quot;qscore&quot;</span>, <span class="at">showCategory=</span><span class="dv">20</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;ORA barplot - qvalue (top 20)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-211-1.png" width="672" /></p>
</div>
<div id="dotplot" class="section level3 hasAnchor">
<h3>Dotplot<a href="module-8.html#dotplot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Dot plot is very similar to bar plot. It has additional capability to encode another score as dot size.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="module-8.html#cb67-1" tabindex="-1"></a><span class="fu">dotplot</span>(ego.sim, <span class="at">showCategory=</span><span class="dv">20</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;Dotplot for ORA (top 20)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-212-1.png" width="672" /></p>
</div>
<div id="enrichment-map" class="section level3 hasAnchor">
<h3>Enrichment Map<a href="module-8.html#enrichment-map" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Enrichment map organizes enriched terms into a network with edges connecting overlapping gene sets. In this way, mutually overlapping gene sets are tend to cluster together, making it easy to identify functional module. Before making the map, similarity must be calculated. This can be done using <code>pairwise_termsim()</code></p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="module-8.html#cb68-1" tabindex="-1"></a>edo <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(ego.sim)</span>
<span id="cb68-2"><a href="module-8.html#cb68-2" tabindex="-1"></a><span class="fu">emapplot</span>(edo)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;ORA Enrichment Map&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-213-1.png" width="672" /></p>
</div>
<div id="upset-plot" class="section level3 hasAnchor">
<h3>Upset plot<a href="module-8.html#upset-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The upsetplot is for visualizing the complex association between genes and gene sets. It emphasizes the gene overlapping among different gene sets.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="module-8.html#cb69-1" tabindex="-1"></a><span class="fu">upsetplot</span>(edo, <span class="at">n=</span><span class="dv">5</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;ORA upset plot (top 5)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-214-1.png" width="672" /></p>
</div>
<div id="details-about-the-input-arguments-for-enrichgo" class="section level3 hasAnchor">
<h3>Details about the input arguments for <code>enrichGO()</code><a href="module-8.html#details-about-the-input-arguments-for-enrichgo" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><code>gene</code> a vector of entrez gene ID.
<code>OrgDb</code> OrgDb object
<code>keyType</code> keytype of input gene
<code>ont</code> One of BP, MF, and CC subontologies, or ALL for all three
<code>pvalueCutoff</code> adjusted pvalue cutoff on enrichment tests to report
<code>pAdjustMethod</code> one of holm, hochberg, hommel, bonferroni, BH, BY, fdr, none
<code>universe</code> background genes. If missing, the all genes listed in the database (eg TERM2GENE table) will be used as background
<code>qvalueCutoff</code> qvalue cutoff on enrichment tests to report as significant. Tests must pass i) pvalueCutoff on unadjusted pvalues, ii) pvalueCutoff on adjusted pvalues and iii) qvalueCutoff on qvalues to be reported
<code>minGSSize</code> minimal size of genes annotated by Ontology term for testing
<code>maxGSSize</code> maximal size of genes annotated for testing
<code>readable</code> whether mapping gene ID to gene Name</p>
</div>
</div>
<div id="a-note-on-supported-organisms" class="section level2 hasAnchor">
<h2>A note on supported organisms<a href="module-8.html#a-note-on-supported-organisms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>GO analyses in <code>clusterProfiler</code> support organisms that have an <code>OrgDb</code> object available. <code>OrgDb</code> (organism databases) objects are databases that contain genome annotations and thus, they are best for converting gene IDs or obtaining GO information for current genome builds.A list of organism databases can be found <a href="https://www.bioconductor.org/packages/release/BiocViews.html#___OrgDb">here</a></p>
</div>
<div id="exercise-2a-gene-set-enrichment-analysis" class="section level2 hasAnchor">
<h2>Exercise 2a: Gene set enrichment analysis<a href="module-8.html#exercise-2a-gene-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-for-running-gene-set-enrichment-analysis-in-clusterprofiler" class="section level3 hasAnchor">
<h3>Data for running gene set enrichment analysis in <code>clusterProfiler</code><a href="module-8.html#data-for-running-gene-set-enrichment-analysis-in-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>GSEA analysis requires a ranked gene list, which contains three features:</p>
<ul>
<li>numeric vector: fold change or other type of numerical variable</li>
<li>named vector: every number has a name, the corresponding gene ID</li>
<li>sorted vector: number should be sorted in decreasing order</li>
</ul>
<p>Since <code>geneList</code> is already in the desired format, we will use it for this exercise. If you havent loaded it, use the command below to import the data. Please see the above section Data for enrichment using <code>clusterProfiler</code> for details regarding the dataset.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="module-8.html#cb70-1" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb70-2"><a href="module-8.html#cb70-2" tabindex="-1"></a><span class="fu">head</span>(geneList)</span></code></pre></div>
<pre><code>##     4312     8318    10874    55143    55388      991 
## 4.572613 4.514594 4.418218 4.144075 3.876258 3.677857</code></pre>
</div>
<div id="perform-go-gene-set-enrichment-analysis" class="section level3 hasAnchor">
<h3>Perform GO gene set enrichment analysis<a href="module-8.html#perform-go-gene-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>clusterProfiler</code> package provides the <code>gseGO()</code> function for gene set enrichment analysis using gene ontology. You can run GSEA as below:</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="module-8.html#cb72-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb72-2"><a href="module-8.html#cb72-2" tabindex="-1"></a>egsea <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> geneList,</span>
<span id="cb72-3"><a href="module-8.html#cb72-3" tabindex="-1"></a>               <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb72-4"><a href="module-8.html#cb72-4" tabindex="-1"></a>               <span class="at">ont          =</span> <span class="st">&quot;ALL&quot;</span>,</span>
<span id="cb72-5"><a href="module-8.html#cb72-5" tabindex="-1"></a>               <span class="at">minGSSize    =</span> <span class="dv">100</span>,</span>
<span id="cb72-6"><a href="module-8.html#cb72-6" tabindex="-1"></a>               <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb72-7"><a href="module-8.html#cb72-7" tabindex="-1"></a>               <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb72-8"><a href="module-8.html#cb72-8" tabindex="-1"></a>               <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb72-9"><a href="module-8.html#cb72-9" tabindex="-1"></a>               <span class="at">eps =</span> <span class="dv">0</span>,</span>
<span id="cb72-10"><a href="module-8.html#cb72-10" tabindex="-1"></a>               <span class="at">verbose      =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="results-of-go-gene-set-enrichment-analysis" class="section level3 hasAnchor">
<h3>Results of GO gene set enrichment analysis<a href="module-8.html#results-of-go-gene-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Examine the results. Do you notice any similarities or differences between this output format and your results from <strong>Module 2</strong> <code>GSEA</code>?</p>
<p>The output table is stored in <code>egsea@result</code>. In this example, 512 processes were significantly enriched.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="module-8.html#cb73-1" tabindex="-1"></a><span class="fu">head</span>(egsea)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID                          Description setSize
## GO:0007059       BP GO:0007059               chromosome segregation     328
## GO:0098813       BP GO:0098813       nuclear chromosome segregation     244
## GO:0000819       BP GO:0000819         sister chromatid segregation     191
## GO:0000070       BP GO:0000070 mitotic sister chromatid segregation     158
## GO:0051276       BP GO:0051276              chromosome organization     432
## GO:0000280       BP GO:0000280                     nuclear division     331
##            enrichmentScore      NES       pvalue     p.adjust       qvalue rank
## GO:0007059       0.5825986 2.738132 2.421350e-31 3.503693e-28 2.123142e-28 1374
## GO:0098813       0.6294381 2.875066 4.565594e-29 3.303208e-26 2.001653e-26  449
## GO:0000819       0.6552843 2.920205 1.633777e-27 7.880251e-25 4.775215e-25  449
## GO:0000070       0.6793857 2.966631 3.166104e-26 1.145338e-23 6.940433e-24  532
## GO:0051276       0.5044982 2.440786 9.538621e-26 2.760477e-23 1.672773e-23 1374
## GO:0000280       0.5436261 2.556505 3.995643e-25 9.636158e-23 5.839246e-23 1246
##                              leading_edge
## GO:0007059 tags=27%, list=11%, signal=24%
## GO:0098813  tags=23%, list=4%, signal=22%
## GO:0000819  tags=25%, list=4%, signal=24%
## GO:0000070  tags=28%, list=4%, signal=28%
## GO:0051276 tags=23%, list=11%, signal=21%
## GO:0000280 tags=26%, list=10%, signal=24%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     core_enrichment
## GO:0007059                                              55143/991/9493/1062/4605/9133/10403/7153/23397/259266/9787/11065/55355/220134/51203/22974/10460/4751/79019/55839/983/4085/81930/81620/332/3832/7272/64151/9212/9319/9055/3833/146909/6790/891/24137/9232/9928/11004/990/5347/29127/26255/701/11130/79682/57405/10615/1894/2491/9700/898/56992/11339/4288/54801/9134/29899/9918/699/6491/1063/26271/55055/54892/29901/79980/9735/55732/81624/23310/292/10051/1104/5359/83990/5885/11200/203068/84722/51115/7283/51647/54908/10592/6732/54069
## GO:0098813                                                                                                                                                                                                                                     55143/991/9493/1062/4605/9133/10403/7153/23397/259266/9787/11065/220134/51203/22974/10460/4751/983/4085/81930/81620/332/3832/7272/64151/9212/9319/9055/3833/146909/6790/891/24137/9232/9928/11004/990/5347/29127/26255/701/11130/57405/10615/1894/2491/9700/898/56992/9134/9918/699/1063/26271/55055
## GO:0000819                                                                                                                                                                                                                                                                               55143/991/9493/1062/4605/10403/7153/23397/9787/11065/220134/51203/22974/10460/4751/983/4085/81930/81620/332/3832/7272/64151/9212/9319/9055/3833/146909/891/24137/9928/11004/990/5347/29127/701/11130/57405/10615/2491/9700/56992/9918/699/1063/26271/55055
## GO:0000070                                                                                                                                                                                                                                                                                        55143/991/9493/1062/4605/10403/23397/9787/11065/220134/51203/22974/4751/983/4085/81930/81620/332/3832/7272/64151/9212/9319/9055/3833/146909/891/24137/9928/11004/5347/29127/701/11130/57405/10615/2491/9700/56992/9918/699/1063/26271/55055/54892
## GO:0051276 55143/991/9493/1062/4605/10403/7153/23397/9787/11065/55355/220134/51203/22974/10460/4751/55839/983/4085/81930/81620/332/3832/7272/64151/9212/9319/9055/3833/146909/891/24137/9232/9928/11004/990/5347/29127/26255/701/9156/11130/57405/10615/79075/2491/9700/5888/898/3149/56992/11339/9134/2237/22948/9918/1058/2643/699/4609/1063/5111/9401/26271/55055/641/1763/54892/8357/3024/3148/3006/4436/9735/908/23310/5499/8607/3008/10051/10576/1104/3009/9631/83990/5885/11200/2072/84722/51115/7283/4678/5588/54908/10592/51377/4683/54069
## GO:0000280                                                                55143/991/9493/1062/4605/9133/10403/7153/23397/259266/9787/11065/220134/51203/22974/4751/10635/27338/983/4085/81930/81620/332/3832/7272/64151/9212/1111/9319/9055/3833/146909/6790/891/24137/9232/9928/1164/11004/4603/5347/29127/26255/701/11130/57405/10615/2491/8438/9700/5888/898/56992/4288/9134/2175/91646/994/9918/699/1063/26271/55055/54892/5902/9585/9735/23310/2253/8877/9088/995/10051/79703/1104/79866/83990/5885/11200/2072/940/1761/84722/51115/7283/80010</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="module-8.html#cb75-1" tabindex="-1"></a><span class="fu">nrow</span>(egsea<span class="sc">@</span>result)</span></code></pre></div>
<pre><code>## [1] 509</code></pre>
</div>
<div id="input-options-for-gsego" class="section level3 hasAnchor">
<h3>Input options for <code>gseGO()</code><a href="module-8.html#input-options-for-gsego" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="rmd-note">
<ul>
<li>
Note that only gene sets having the size within
[<code>minGSSize</code>, <code>maxGSSize</code>] will be tested.
</li>
<li>
Similar to <code>enrichGO()</code>, you can specify subontology
using the argument <code>ont</code>. It takes one option among - BP,
MF, CC or ALL for biological process, molecular function, cellular
co-localization or all subontologies respectively
</li>
<li>
<code>pvalueCutoff</code> defines the cutoff for pvalue that is used
for determining significant processes
</li>
<li>
Setting <code>eps</code> to zero improves estimation.
</li>
<li>
<code>pAdjustMethod</code> can be one of holm, hochberg,
hommel, bonferroni, BH, BY, fdr, none
</li>
</ul>
</div>
</div>
<div id="details-about-the-input-arguments-for-gsego" class="section level3 hasAnchor">
<h3>Details about the input arguments for <code>gseGO()</code><a href="module-8.html#details-about-the-input-arguments-for-gsego" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><code>geneList</code> order ranked geneList
<code>ont</code> one of BP, MF, and CC subontologies, or ALL for all three
<code>OrgDb</code> OrgDb
<code>keyType</code> keytype of gene
<code>exponent</code> weight of each step
<code>minGSSize</code> minimal size of each geneSet for analyzing
<code>maxGSSize</code> maximal size of genes annotated for testing
<code>eps</code> This parameter sets the boundary for calculating the p value
<code>pvalueCutoff</code> pvalue Cutoff
<code>pAdjustMethod</code> <code>pAdjustMethod</code> one of holm, hochberg, hommel, bonferroni, BH, BY, fdr, none
<code>verbose</code> print message or not
<code>seed</code> logical
<code>by</code> one of fgsea or DOSE</p>
</div>
</div>
<div id="exercise-2b.-visualise-the-results-of-gene-set-enrichment-analysis" class="section level2 hasAnchor">
<h2>Exercise 2b. Visualise the results of gene set enrichment analysis<a href="module-8.html#exercise-2b.-visualise-the-results-of-gene-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="dotplot-1" class="section level3 hasAnchor">
<h3>Dotplot<a href="module-8.html#dotplot-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You can use the function <code>dotplot()</code> to summarise GSEA results.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="module-8.html#cb77-1" tabindex="-1"></a><span class="fu">dotplot</span>(egsea, <span class="at">showCategory=</span><span class="dv">20</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;Dotplot for GSEA (top 20)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-220-1.png" width="672" /></p>
</div>
<div id="ridgeline-plot" class="section level3 hasAnchor">
<h3>Ridgeline plot<a href="module-8.html#ridgeline-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The function <code>ridgeplot()</code> will visualize expression distributions of core enriched genes for GSEA enriched categories. It helps you to interpret up/down-regulated pathways.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="module-8.html#cb78-1" tabindex="-1"></a>enrichplot<span class="sc">::</span><span class="fu">ridgeplot</span>(egsea, <span class="at">showCategory =</span> <span class="dv">20</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;Ridgeplot for GSEA (top 20)&quot;</span>)</span></code></pre></div>
<pre><code>## Picking joint bandwidth of 0.285</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-221-1.png" width="672" /></p>
</div>
<div id="running-score-and-preranked-list-of-gsea-result" class="section level3 hasAnchor">
<h3>Running score and preranked list of GSEA result<a href="module-8.html#running-score-and-preranked-list-of-gsea-result" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Running score and preranked list are traditional methods for visualizing GSEA result. You are familiar with these visualisations from Module 2. The function <code>gseaplot()</code> supports visualising both the distribution of the gene set and the enrichment score.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="module-8.html#cb80-1" tabindex="-1"></a><span class="fu">gseaplot</span>(egsea, <span class="at">geneSetID =</span> <span class="dv">1</span>, <span class="at">by =</span> <span class="st">&quot;runningScore&quot;</span>, <span class="at">title =</span> egsea<span class="sc">$</span>Description[<span class="dv">1</span>])</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-222-1.png" width="672" /></p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="module-8.html#cb81-1" tabindex="-1"></a><span class="fu">gseaplot</span>(egsea, <span class="at">geneSetID =</span> <span class="dv">1</span>, <span class="at">by =</span> <span class="st">&quot;preranked&quot;</span>, <span class="at">title =</span> egsea<span class="sc">$</span>Description[<span class="dv">1</span>])</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-223-1.png" width="672" /></p>
<p>Another method to plot GSEA result is the <code>gseaplot2</code> function:</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="module-8.html#cb82-1" tabindex="-1"></a><span class="fu">gseaplot2</span>(egsea, <span class="at">geneSetID =</span> <span class="dv">1</span>, <span class="at">title =</span> egsea<span class="sc">$</span>Description[<span class="dv">1</span>])</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-224-1.png" width="672" /></p>
</div>
<div id="enrichment-map-1" class="section level3 hasAnchor">
<h3>Enrichment Map<a href="module-8.html#enrichment-map-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The function <code>emapplot</code> also supports visualising results of GSEA. As we did before, let us first calculate similarity using <code>pairwise_termsim()</code></p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="module-8.html#cb83-1" tabindex="-1"></a>edo2 <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(egsea)</span>
<span id="cb83-2"><a href="module-8.html#cb83-2" tabindex="-1"></a><span class="fu">emapplot</span>(edo2)<span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;GSEA Enrichment Map&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-225-1.png" width="672" /></p>
</div>
</div>
<div id="what-next" class="section level2 hasAnchor">
<h2>What next?<a href="module-8.html#what-next" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This figure gives a complete overview of functionalities of <code>clusterProfiler</code></p>
<div class="float">
<img src="Module5/clusterprofiler/images/clusterProfiler-functions.png" alt="1" />
<div class="figcaption">1</div>
</div>
</div>
<div id="explore-other-features-of-clusterprofiler" class="section level2 hasAnchor">
<h2>Explore other features of <code>clusterProfiler</code><a href="module-8.html#explore-other-features-of-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For other functionalities in <code>clusterProfiler</code> please refer to detailed examples in this <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/index.html">book</a></p>
<div id="bonus---try-it-yourself" class="section level3 hasAnchor">
<h3>Bonus - Try it yourself:<a href="module-8.html#bonus---try-it-yourself" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="rmd-bonus">
<p>
Using your knowledge of <code>clusterProfiler</code>, write scripts
to perform the following analysis. Use the <code>geneList</code>
dataset.
</p>
<ul>
<li>
Run ORA against <strong>GO molecular function</strong> by converting
<code>gene</code> to <strong>uniprot IDs</strong>
</li>
<li>
Run ORA against <strong>KEGG pathways</strong>,
<strong>Reactome</strong> and <strong>Wikipathways</strong>
databases
</li>
<li>
Run GSEA against <strong>KEGG pathways</strong>,
<strong>Reactome</strong> and <strong>Wikipathways</strong>
databases
</li>
<li>
Use your data to run ORA and GSEA using
<code>clusterProfiler</code>
</li>
</ul>
<p>
Hint: <code>clusterProfiler</code> provides different functions for
testing against multiple databases. Refer the <a
href="https://yulab-smu.top/biomedical-knowledge-mining-book/index.html">book</a>
for complete list.
</p>
</div>
</div>
<div id="ontologies-and-pathway-databases-supported-by-clusterprofiler" class="section level3 hasAnchor">
<h3>Ontologies and pathway databases supported by <code>clusterProfiler</code><a href="module-8.html#ontologies-and-pathway-databases-supported-by-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Disease Ontology (via <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>)</li>
<li><a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> (via <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>)</li>
<li><a href="http://www.disgenet.org/web/DisGeNET/menu/home">DisGeNET</a> (via <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>)</li>
<li>Gene Ontology (supports many species with GO annotation query online via <a href="https://bioconductor.org/packages/AnnotationHub/">AnnotationHub</a>)</li>
<li>KEGG Pathway and Module with latest online data (supports more than 4000 species listed in <a href="http://www.genome.jp/kegg/catalog/org_list.html" class="uri">http://www.genome.jp/kegg/catalog/org_list.html</a>)</li>
<li>Reactome Pathway (via <a href="https://www.bioconductor.org/packages/ReactomePA">ReactomePA</a>)</li>
<li>DAVID (via <a href="https://www.bioconductor.org/packages/RDAVIDWebService">RDAVIDWebService</a>)</li>
<li><a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a>
<ul>
<li>hallmark gene sets</li>
<li>positional gene sets</li>
<li>curated gene sets</li>
<li>motif gene sets</li>
<li>computational gene sets</li>
<li>GO gene sets</li>
<li>oncogenic signatures</li>
<li>immunologic signatures</li>
</ul></li>
<li>Other Annotations
<ul>
<li>from other sources (e.g.<a href="http://www.disgenet.org/web/DisGeNET/menu/home">DisGeNET</a> as <a href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">an example</a>)</li>
<li>users annotation</li>
<li>customized ontology</li>
<li>and many others</li>
</ul></li>
</ul>
</div>
<div id="all-publications-describing-clusterprofiler-can-be-found-here" class="section level3 hasAnchor">
<h3>All publications describing <code>clusterProfiler</code> can be found here:<a href="module-8.html#all-publications-describing-clusterprofiler-can-be-found-here" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>T Wu<sup>#</sup>, E Hu<sup>#</sup>, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. <strong><em>The Innovation</em></strong>. 2021, 2(3):100141.
doi: <a href="https://doi.org/10.1016/j.xinn.2021.100141">10.1016/j.xinn.2021.100141</a></li>
<li><strong>G Yu</strong><sup>*</sup>. Gene Ontology Semantic Similarity Analysis Using GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks. <strong><em>Methods in Molecular Biology</em></strong>. 2020, 2117:207-215. Humana, New York, NY.
doi: <a href="https://doi.org/10.1007/978-1-0716-0301-7_11">10.1007/978-1-0716-0301-7_11</a></li>
<li><strong>G Yu</strong><sup>*</sup>. Using meshes for MeSH term enrichment and semantic analyses. <strong><em>Bioinformatics</em></strong>. 2018, 34(21):37663767.
doi: <a href="https://doi.org/10.1093/bioinformatics/bty410">10.1093/bioinformatics/bty410</a></li>
<li><strong>G Yu</strong>, QY He<sup>*</sup>. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization. <strong><em>Molecular BioSystems</em></strong>. 2016, 12(2):477-479.
doi: <a href="https://doi.org/10.1039/C5MB00663E">10.1039/C5MB00663E</a></li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, and QY He<sup>*</sup>. ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization. <strong><em>Bioinformatics</em></strong>. 2015, 31(14):2382-2383.
doi: <a href="https://doi.org/10.1093/bioinformatics/btv145">10.1093/bioinformatics/btv145</a></li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, GR Yan, QY He<sup>*</sup>. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. <strong><em>Bioinformatics</em></strong>. 2015, 31(4):608-609.
doi: <a href="https://doi.org/10.1093/bioinformatics/btu684">10.1093/bioinformatics/btu684</a></li>
<li><strong>G Yu</strong>, LG Wang, Y Han and QY He<sup>*</sup>. clusterProfiler: an R package for comparing biological themes among gene clusters. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.
doi: <a href="https://doi.org/10.1089/omi.2011.0118">10.1089/omi.2011.0118</a></li>
<li><strong>G Yu</strong>, F Li, Y Qin, X Bo<sup>*</sup>, Y Wu, S Wang<sup>*</sup>. GOSemSim: an R package for measuring semantic similarity among GO terms and gene products. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.
doi: <a href="https://doi.org/10.1093/bioinformatics/btq064">10.1093/bioinformatics/btq064</a></li>
</ol>

</div>
</div>
</div>
</div>
</div>
</div>






            </section>

          </div>
        </div>
      </div>
<a href="module-7.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/bioinformaticsdotca/PNA_2024/blob/main/080-module-8.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

</body>

</html>
